% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,openany,oneside]{sphinxmanual}
\usepackage[utf8]{inputenc}
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage[english]{babel}
\usepackage{times}
\usepackage[Bjarne]{fncychap}
\usepackage{longtable}
\usepackage{sphinx}
\usepackage{multirow}
\usepackage{eqparbox}
\usepackage{xspace}
\usepackage[version=3]{mhchem}

%% BIBLATEX uses BIBER
%% For Ubuntu install: apt-get install biber
%% compile pdflatex biber pdflatex pdflatex
\usepackage[backend=biber,natbib,style=nature,date=year,isbn=false,%
  doi=false,url=false]{biblatex}
\addbibresource{bibtex/group-literature.bib}

\graphicspath{{./graphics/}}

%\newcommand{\todo}[1]{\xspace{\bfseries\sffamily\textcolor{red}{[#1]}}\xspace}
\newcommand{\todo}[1]{}

\newcommand\etal{\emph{et~al.\@}\xspace}
\newcommand\gui[1]{\textsl{\guilsinglleft#1\guilsinglright\xspace}}

\hyphenation{Meth-od-ol high-light-ed}

\addto\captionsenglish{\renewcommand{\figurename}{Fig.\@ }}
\addto\captionsenglish{\renewcommand{\tablename}{Table }}
\SetupFloatingEnvironment{literal-block}{name=Listing }

\addto\extrasenglish{\def\pageautorefname{page}}

\setcounter{tocdepth}{1}


\def\tightlist{}

\title{SIRIUS Documentation}
\date{\today}
\release{\IfFileExists{./version.txt}{\input{./version.txt}}{3}}
\author{Sebastian Böcker, Kai Dührkop, \\ Markus Fleischauer, Marcus Ludwig}
\newcommand{\sphinxlogo}{\includegraphics{sirius.png}\par}
\renewcommand{\releasename}{Release}
\makeindex

\makeatletter
\def\PYG@reset{\let\PYG@it=\relax \let\PYG@bf=\relax%
    \let\PYG@ul=\relax \let\PYG@tc=\relax%
    \let\PYG@bc=\relax \let\PYG@ff=\relax}
\def\PYG@tok#1{\csname PYG@tok@#1\endcsname}
\def\PYG@toks#1+{\ifx\relax#1\empty\else%
    \PYG@tok{#1}\expandafter\PYG@toks\fi}
\def\PYG@do#1{\PYG@bc{\PYG@tc{\PYG@ul{%
    \PYG@it{\PYG@bf{\PYG@ff{#1}}}}}}}
\def\PYG#1#2{\PYG@reset\PYG@toks#1+\relax+\PYG@do{#2}}

\expandafter\def\csname PYG@tok@gd\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.63,0.00,0.00}{##1}}}
\expandafter\def\csname PYG@tok@gu\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.50,0.00,0.50}{##1}}}
\expandafter\def\csname PYG@tok@gt\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.27,0.87}{##1}}}
\expandafter\def\csname PYG@tok@gs\endcsname{\let\PYG@bf=\textbf}
\expandafter\def\csname PYG@tok@gr\endcsname{\def\PYG@tc##1{\textcolor[rgb]{1.00,0.00,0.00}{##1}}}
\expandafter\def\csname PYG@tok@cm\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@vg\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@vi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@mh\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@cs\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}\def\PYG@bc##1{\setlength{\fboxsep}{0pt}\colorbox[rgb]{1.00,0.94,0.94}{\strut ##1}}}
\expandafter\def\csname PYG@tok@ge\endcsname{\let\PYG@it=\textit}
\expandafter\def\csname PYG@tok@vc\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@il\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@go\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.20,0.20,0.20}{##1}}}
\expandafter\def\csname PYG@tok@cp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@gi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.63,0.00}{##1}}}
\expandafter\def\csname PYG@tok@gh\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.00,0.50}{##1}}}
\expandafter\def\csname PYG@tok@ni\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.84,0.33,0.22}{##1}}}
\expandafter\def\csname PYG@tok@nl\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.13,0.44}{##1}}}
\expandafter\def\csname PYG@tok@nn\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\expandafter\def\csname PYG@tok@no\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.38,0.68,0.84}{##1}}}
\expandafter\def\csname PYG@tok@na\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@nb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@nc\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\expandafter\def\csname PYG@tok@nd\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.33,0.33,0.33}{##1}}}
\expandafter\def\csname PYG@tok@ne\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@nf\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.49}{##1}}}
\expandafter\def\csname PYG@tok@si\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.44,0.63,0.82}{##1}}}
\expandafter\def\csname PYG@tok@s2\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@nt\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.45}{##1}}}
\expandafter\def\csname PYG@tok@nv\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@s1\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@ch\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@m\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@gp\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\expandafter\def\csname PYG@tok@sh\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@ow\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@sx\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\expandafter\def\csname PYG@tok@bp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@c1\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@o\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PYG@tok@kc\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@c\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@mf\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@err\endcsname{\def\PYG@bc##1{\setlength{\fboxsep}{0pt}\fcolorbox[rgb]{1.00,0.00,0.00}{1,1,1}{\strut ##1}}}
\expandafter\def\csname PYG@tok@mb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@ss\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.32,0.47,0.09}{##1}}}
\expandafter\def\csname PYG@tok@sr\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.14,0.33,0.53}{##1}}}
\expandafter\def\csname PYG@tok@mo\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@kd\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@mi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@kn\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@cpf\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@kr\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@s\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@kp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@w\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.73,0.73}{##1}}}
\expandafter\def\csname PYG@tok@kt\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.56,0.13,0.00}{##1}}}
\expandafter\def\csname PYG@tok@sc\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@sb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@k\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@se\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@sd\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}

\def\PYGZbs{\char`\\}
\def\PYGZus{\char`\_}
\def\PYGZob{\char`\{}
\def\PYGZcb{\char`\}}
\def\PYGZca{\char`\^}
\def\PYGZam{\char`\&}
\def\PYGZlt{\char`\<}
\def\PYGZgt{\char`\>}
\def\PYGZsh{\char`\#}
\def\PYGZpc{\char`\%}
\def\PYGZdl{\char`\$}
\def\PYGZhy{\char`\-}
\def\PYGZsq{\char`\'}
\def\PYGZdq{\char`\"}
\def\PYGZti{\char`\~}
% for compatibility with earlier versions
\def\PYGZat{@}
\def\PYGZlb{[}
\def\PYGZrb{]}
\makeatother

\renewcommand\PYGZsq{\textquotesingle}
        
\begin{document}

\maketitle
{\huge\textcolor{red}{ This manual is currently being updated to describe 
all new features of SIRIUS 4.0.}}

\tableofcontents
\phantomsection\label{index::doc}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%  CHAPTER  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\chapter{Introduction}
\label{introduction:introduction}
\label{introduction::doc}
\label{introduction:welcome-to-sirius-s-documentation}

\section{Welcome}

SIRIUS is a \emph{Java} software for analyzing metabolites from tandem mass
spectrometry data. It combines the analysis of isotope patterns in MS spectra
with the analysis of fragmentation patterns in MS/MS spectra, and uses
CSI:FingerID as a web service to search in molecular structure databases.

SIRIUS requires \textbf{high mass accuracy} data. The mass deviation of
your MS and MS/MS spectra should be within 20 ppm.  Mass Spectrometry
instruments such as TOF, Orbitrap and FT-ICR usually provide high mass
accuracy data, as well as coupled instruments like Q-TOF, IT-TOF or
IT-Orbitrap.  Spectra measured with a quadrupole or linear trap do not
provide the high mass accuracy that is required for our method.  See
Sec.~\ref{sec:mass-deviations} on what ``mass accuracy'' means in detail for
SIRIUS.

SIRIUS expects \textbf{MS and MS/MS} spectra as input.  It is possible to
omit the MS data, but it will make the analysis more time consuming and
might give you worse results.  In this case, you should consider limiting the
candidate molecular formulas to those found in PubChem.

SIRIUS expects \textbf{processed peak lists} (centroided spectra). It does
not contain routines for peak picking from profiled spectra, nor routines for
merging spectra in an LC/MS run.  This is a deliberate design decision: We
want you to use the best peak picking software out there --- or
alternatively, your favorite software.  There are several tools specialized
for this task, such as OpenMS, MZmine or XCMS.  The Trinity workflow
(see \url{https://www.youtube.com/watch?v=zDcY7iuvyQY}) produces picked peak
lists that can directly be imported into SIRIUS.

SIRIUS will identify the molecular formula of the measured precursor ion,
and will also annotates the spectrum by providing a molecular formula for
each fragment peak.  Peaks that receive no annotation are assumed to be noise
peaks.  Furthermore, a \textbf{fragmentation tree} is predicted; this tree
contains the predicted fragmentation reaction leading to the fragment peaks.

SIRIUS uses CSI:FingerID to identify the structure of a compound by
searching in a molecular structure database.  Here, ``structure'' refers to
``conformation'': 

SIRIUS can be used within an analysis pipeline. For example, you can
identify the molecular formula of the ion and the fragment peaks, and use
this information as input for other tools such as FingerID or MAGMa to
identify the structure of the measured compound. For this purpose, you can
also use the SIRIUS library directly, instead of the command line
interface. See {\hyperref[library::doc]{\crossref{\DUrole{doc}{SIRIUS Java
Library}}}}.

Since version~3.1, our software ships with a \textbf{Graphical User
Interface} (GUI).  A version without GUI is available as separate download.

\clearpage

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Literature}
\label{introduction:literature}

The \emph{scientific development} behind SIRIUS and CSI:FingerID required
numerous man-years of PhD students, postdocs and principal investigators; an
educated guess would be roughly 30 man-years.  This estimate does not include
building the shiny Graphical User Interface that was introduced in
version~3.1.  But it is not the user interface or software development that
does the work here; it is our scientific research that made SIRIUS and
CSI:FingerID possible.  It is understood that the work of 10 years cannot be
described in a single paper.

Please cite all papers that you feel relevant for your work.  Please do not
cite this manual or the SIRIUS or CSI:FingerID website, but rather our
scientific papers.

\subsection{Searching in molecular structure databases}
\label{introduction:searching-in-molecular-structure-databases}

\begin{itemize}
\item Kai Dührkop, Huibin Shen, Marvin Meusel, Juho Rousu and Sebastian Böcker.
\textbf{Searching molecular structure databases with tandem mass spectra using CSI:FingerID.}
\emph{Proc Natl Acad Sci U S A}, 112(41):12580--12585, 2015. 

\item Huibin Shen, Kai Dührkop, Sebastian Böcker and Juho Rousu.
\textbf{Metabolite Identification through Multiple Kernel Learning on Fragmentation Trees.}
\emph{Bioinformatics}, 30(12):i157--i164, 2014. Proc. of \emph{Intelligent Systems for Molecular Biology} (ISMB 2014).
\end{itemize}

\subsection{Fragmentation Tree Computation}
\label{introduction:fragmentation-tree-computation}

\begin{itemize}
\item Sebastian Böcker and Kai Dührkop.
\textbf{Fragmentation trees reloaded.}
\emph{J Cheminform}, 8:5, 2016.

\item W. Timothy J. White, Stephan Beyer, Kai Dührkop, Markus Chimani and Sebastian Böcker.
\textbf{Speedy Colorful Subtrees.}
In \emph{Proc. of Computing and Combinatorics Conference} (COCOON 2015), volume 9198 of \emph{Lect Notes Comput Sci}, pages 310--322. Springer, Berlin, 2015.

\item Imran Rauf, Florian Rasche, François Nicolas and Sebastian Böcker.
\textbf{Finding Maximum Colorful Subtrees in practice.}
\emph{J Comput Biol}, 20(4):1--11, 2013.

\item Florian Rasche, Aleš Svatoš, Ravi Kumar Maddula, Christoph Böttcher and Sebastian Böcker.
\textbf{Computing fragmentation trees from tandem mass spectrometry data.}
\emph{Anal Chem} 83(4):1243–1251, 2011.

\item Sebastian Böcker and Florian Rasche.
\textbf{Towards de novo identification of metabolites by analyzing tandem mass spectra.}
\emph{Bioinformatics} 24(16):i49--i55, 2008.
\end{itemize}


\subsection{Isotope pattern analysis}
\label{introduction:isotope-pattern-analysis}

\begin{itemize}
\item Sebastian Böcker, Matthias C. Letzel, Zsuzsanna Lipták and Anton
Pervukhin.
\textbf{SIRIUS: decomposing isotope patterns for metabolite identification.}
\emph{Bioinformatics} 25(2): 218--224, 2009.

\item Sebastian Böcker, Matthias Letzel, Zsuzsanna Lipták and Anton Pervukhin.
\textbf{Decomposing metabolomic isotope patterns.}
In \emph{Proc. of Workshop on Algorithms in Bioinformatics} (WABI 2006), volume 4175 of \emph{Lect Notes Comput Sci}, pages 12--23. Springer, Berlin, 2006. 
\end{itemize}


\subsection{Auto-detection of elements}
\label{introduction:auto-detection-of-elements}

\begin{itemize}
\item Marvin Meusel, Franziska Hufsky, Fabian Panter, Daniel Krug, Rolf Müller and Sebastian Böcker.
\textbf{Predicting the presence of uncommon elements in unknown biomolecules from isotope patterns.}
\emph{Anal Chem}, 88(15):7556--7566, 2016. 
\end{itemize}

\subsection{Mass decomposition}
\label{introduction:mass-decomposition}

\begin{itemize}
\item Kai Dührkop, Marcus Ludwig, Marvin Meusel and Sebastian Böcker.
\textbf{Faster mass decomposition.}
In \emph{Proc. of Workshop on Algorithms in Bioinformatics} (WABI 2013), volume 8126 of \emph{Lect Notes Comput Sci}, pages 45--58. Springer, Berlin, 2013.

\item Sebastian Böcker and Zsuzsanna Lipták.
\textbf{A fast and simple algorithm for the Money Changing Problem.}
\emph{Algorithmica} 48(4):413--432, 2007.

\item Sebastian Böcker and Zsuzsanna Lipták.
\textbf{Efficient Mass Decomposition.}
In \emph{Proc. of ACM Symposium on Applied Computing} (ACM SAC 2005), pages 151-157. ACM press, New York, 2005. 
\end{itemize}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%  CHAPTER  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\chapter{Installation}
\label{install:installation}\label{install::doc}\label{install:user-interface}

To install SIRIUS, you only have to extract the archive you have downloaded
to an arbitrary directory where you have write permissions.  To run SIRIUS,
you need to have a \textbf{Java Runtime Environment (JRE) version 7 or
higher} installed.  \textbf{If you run a 64-bit operating system, you also
have to install a 64-bit Java runtime environment!}  If you have trouble
installing SIRIUS, please let us know at \url{sirius@uni-jena.de} and we will
see if we can help.  If you find that our installation guide is incomplete,
or if you have some tricks that you want to share with your fellow
scientists, please let us know so we can include them in this manual.

\textbf{Warning!}  All advice given here on how to get SIRIUS running on your
system, is given without any warranty!  If you are not sure what you are
doing, you might want to contact someone who does.  (Remember, last Friday in
July is System Administrator Appreciation Day!)


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Windows}
\label{install:windows}

Extract the archive to an arbitrary directory where you have write
permissions, such as \verb|C:\SIRIUS|.

To check if you have a 64-bit Windows operating system, open the Windows
settings, go to ``System'', then ``Info''.  You will see a line ``System
type'' that tells you if your operating system is 32- or 64-bit.  To check if
your Java version is 64-bit, open a Command Prompt (press the Windows key,
then start typing ``command prompt'', start the Command Prompt) and
type \code{java -version}.  If Windows complains that the java command is
unknown, type
\verb!cd C:\ProgramData\Oracle\Java\javapath!,
then \code{java -version}.  In the last line of the output, you will see if
you have 64-bit Java installed.  If this is not the case, but your Windows is
64-bit, you have to install the correct Java version
from \url{https://java.com/}.  Press ``Download'', but \emph{do not use the
big red button to download Java!}  Instead, go to ``See all Java Downloads'',
then choose ``Windows 64-bit''.

Go to the SIRIUS directory.  Run \code{sirius-gui.exe} to start the
graphical user interface.  You might want to create a link on your desktop:
Click and drag the file \code{sirius-gui.exe} to the desktop, keeping the
ALT key pressed.  You can rename the link on your desktop as you like.  You
start SIRIUS by double-clicking this link.

Run \code{sirius-console.exe} for the SIRIUS command line tool.  To
execute the SIRIUS command line tool from every location on your system,
you have to add the location of the \code{sirius-console.exe} to your PATH
environment variable: Open the Windows Setting, type ``advanced'' in the
search window, say ``yes'' if Windows asks you.  Press the ``Environment
Variables'' button, select the ``Path'' variable in the lower panel, press
``Edit\dots'', press ``New'', enter the full directory path of SIRIUS,
press RETURN.  Close the Command Prompt, open a new one,
type \code{sirius-console-64.exe}.

\paragraph{Frequently asked questions}

\begin{itemize}
\item ``Is running Java a security problem?''  $\to$ Only if you enable
  execution of Java programs in your web browser. Today, there is practically
  no need to run Java from the browser, so you should disable this option (it
  is probably disabled already).  Always keep your JRE up-to-date.

\item ``I am running Windows and I get an error message, `ERROR: JAVA\_HOME is
  set to an invalid directory'.'' $\to$ Probably, deleting the JAVA\_HOME
  environment variable will solve the problem, see the instructions for the
  PATH variable above.

\item ``I want to run SIRIUS on Windows~XP!''  $\to$ Don't.  Running
  Windows~XP on a computer connected to the internet is a security hazzard.
\end{itemize}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Linux and Mac OS\,X}
\label{install:linux-and-macosx}

For Mac OS\,X, make sure that you have installed JAVA version~8 or higher
from \url{https://java.com/}; Apple only provides the deprecated version~6.

Extract SIRIUS into a folder of your choice, for example \code{\textasciitilde{}/sirius}.

To start the SIRIUS command line tool you then have to enter:
\begin{Verbatim}[commandchars=\\\{\}]
\textasciitilde{}/sirius/bin/sirius
\end{Verbatim}

To start the graphical user interface of SIRIUS:
\begin{Verbatim}[commandchars=\\\{\}]
\textasciitilde{}/sirius/bin/sirius-gui
\end{Verbatim}

To execute SIRIUS from every location you have to add 
\code{/path/to/sirius/bin}
to your PATH variable. To do so, open \code{\textasciitilde{}/.bashrc} in an 
editor and add the following line (replacing the placeholder path):
\begin{Verbatim}[commandchars=\\\{\}]
export PATH=\PYGZdl{}PATH:\textasciitilde{}/sirius/bin/
\end{Verbatim}
Note that you have to reopen your ''bash'' shell to make the changes effective.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Proxy servers}
\label{install:proxy}
To use database related functionality of Sirius, it needs an 
Internet connection. You have to ensure that Sirius is not blocked by any 
security software on your computer.

If you have to use a proxy server to connect to the Internet, Sirius 
automatically uses the system wide Java proxy 
configuration\footnote{\sphinxAtStartFootnote{\url{https://www.java.com/de/download/help/proxy_setup.xml}}}
if available. Alternatively you can specify the proxy configuration in the 
Sirius user interface setting (see Section~\ref{sec:settings}). 


If Sirius cannot connect to the Internet, it will report on which stage the
error occurred.

\begin{center}
  \includegraphics[width=0.4\textwidth]{{connectionCheck}.png}
\end{center}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Installing Gurobi and CPLEX}
\label{install:gurobi}

SIRIUS ships with the GLPK (GNU Linear Programming Kit)
\footnote{\sphinxAtStartFootnote\url{https://www.gnu.org/software/glpk/}} 
Integer Linear Program solver which allows us to swiftly compute
fragmentation trees in most cases.  However, if you want to analyze large
molecules and/or spectra with many peaks and/or a large number of spectra,
you can greatly improve running time by using a faster solver.  SIRIUS also
supports Integer Linear Program solvers Gurobi
\footnote{\sphinxAtStartFootnote\url{http://www.gurobi.com/}} 
and CPLEX 
\footnote{\sphinxAtStartFootnote\url{https://www.ibm.com/products/ilog-cplex-optimization-studio}}.
These are commercial solvers which offer a
free academic license for university members.  You can find installation
instruction on their websites. Using Gurobi or CPLEX will greatly improve the
speed of fragmentation tree computations, which is the most time-intense step
of the computational analysis.  Beside this, there will be no differences in
using Gurobi, CPLEX or GLPK.  
To use Gurobi set the environment variable GUROBI\_HOME to a valid Gurobi 
installation location e.g. \code{/path/to/gurobi750/linux64}.
Similarly, to use CPLEX set CPLEX\_HOME to 
\code{/path/to/CPLEX\_Studio/cplex/lib}.
SIRIUS will automatically use Gurobi or CPLEX as its solver if corresponding 
environment variables are specified.
You can manually set the preferred ILP solvers in the settings dialog (GUI).


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Getting started}

\subsection{Working in single mode}
\label{gui:working-in-single-mode}

SIRIUS's ``Single mode'' corresponds to analyzing a single compound with
one or more mass spectra.

\begin{enumerate}
\item {} 
Move the three files \code{txt/chelidonine\_ms.txt}, \code{txt/chelidonine\_msms1.txt} and \code{chelidonine\_msms2.txt} from the demo data via Drag and Drop into the application window

\item {} 
The following dialog offers you to select the columns for mass and intensity values. Just press \gui{Ok} as the default values are already correct.

\item {} 
You see the load dialog with three spectra. The first spectra is wrongly
annotated as \emph{MS/MS} spectrum but should be an \emph{MS1} spectrum
instead. Just select \gui{MS 1} in the drop down list labeled with \gui{ms
level}.

\item {} 
All other options are fine. However, you might want to choose a more memorizable name in the \gui{compound name} field.

\item {} 
Press the \gui{OK} button. The newly imported compound should now appear in your compound list on the left side.

\item {} 
Choose the compound, right-click on it and press \gui{Compute}.

\item {} 
In the compute dialog all options should be fine. Just check that the correct parent mass is chosen. You might want to add Chlorine or Fluorine to the set of considered elements. Furthermore, you can change the instrument type to \gui{Orbitrap}

\item {} 
Just look into the candidate list: The first molecular formula has a quite large score. Furthermore, the second molecular formula has a much lower score. This is a good indication that the identification is correct. However, you can take a look at the fragmentation tree: Do the peak annotation look correct? Take a look at the spectrum view: Are all high intensive peaks are explained?

\item {} 
You can now save the result list as CSV file (by pressing the \gui{Export Results} button). Maybe you want save your workspace, too. Just press the \gui{Save Workspace} button.

\end{enumerate}


\subsection{Working in batch mode}

SIRIUS's ``Batch mode'' corresponds to analyzing many compounds at once,
each having one or more mass spectra.

\begin{enumerate}
\item {} 
Move the files \code{Bicuculline.ms} and \code{Kaempferol.ms} from the demo data via Drag and Drop into the application window

\item {} 
The two compounds are now displayed in the compound list

\item {} 
Just check if the ionization and parent mass is correctly annotated. You can change this values by clicking on the compound and then on \gui{Edit}.

\item {} 
Click on the \gui{Compute All} button.

\item {} 
You can now select the allowed elements, the instrument type as well as the maximal allowed mass deviation. Be aware that this settings will be used for all imported compounds

\item {} 
Choose \gui{Orbitrap} in the instrument field and press \gui{OK}

\item {} 
A \gui{...} symbol occurs on the lower right corner of each compound. This 
means that the compound will be computed soon. A gear symbol tells you that 
this compound is currently computed in background. A check mark appears in 
all compounds that were successfully computed, a red cross marks compounds 
which computation fails.

\item {} 
Probably you will not see anything than a check mark, as the computation is very fast. However, if you see a compound with a red cross you might want to compute it again in Single Mode. Check if the parent mass and ionization is correct.

\item {} 
Sometimes a computation might take a long time (e.g. for compounds with a lot 
of elements or very high masses). You can cancel the computation of a single 
compound by selecting \gui{Cancel Computation} in the right-click context 
menu. You can cancel the computation of all compounds by clicking on 
\gui{Cancel Computation} in the toolbar.
\end{enumerate}


\subsection{Identifying a CASMI challenge}
\label{gui:identifying-a-casmi-challenge}\begin{enumerate}
\item {} 
Download the files \url{http://casmi-contest.org/2014/Challenge2014/Challenge1/1\_MS.txt} and \url{http://casmi-contest.org/2014/Challenge2014/Challenge1/1\_MSMS.txt}

\item {} 
Move these files via Drag and Drop into the application window

\item {} 
Change the ms level of the first file into \gui{Ms 1}

\item {} 
Click on \gui{OK}

\item {} 
Click on \gui{Compute} in the right-click context menu of the imported compound

\item {} 
Choose \gui{Q-TOF} as instrument and press the \gui{OK} button

\item {} 
\emph{C23H46NO7P} should be suggested as number one hit in the candidate list

\end{enumerate}
  
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%  CHAPTER  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\chapter{Prerequisites}\label{gui::prerequisites}

\section{Spectral quality}

SIRIUS and CSI:FingerID have been trained on a wide variety of data,
including data from different instrument types.  Nevertheless, certain
aspects of the mass spectra are important so that our software can
successfully process your data:
%
\begin{itemize}
\item Be reminded that SIRIUS requires \textbf{high mass accuracy} data:
  The mass deviation should be within 20 ppm.  We are confident that
  SIRIUS can also give useful information for worse mass accuracy (say,
  50~ppm), but you should know what you are doing if you are processing such
  data.

\item It is understood that some molecules generate more fragments, whereas
  others have sparse fragmentation spectra.  But it is also important to
  understand that without sufficient information, it is impossible to deduce
  the structure or even the molecular formula from a tandem mass spectrum
  that contains almost no peaks.  For example, three peaks in a fragmentation
  spectrum measured with 1~ppm mass accuracy contain about 60~bit of
  information, ignoring dependencies between fragments and distibution of
  molecular masses.  With this information, it is simply not possible to find
  the correct structure in a database such as PubChem, containing 100~million
  structures.  In comparison, ten peaks measured with 20~ppm mass accuracy
  contain about 156~bit of information, again ignoring dependencies and
  distributions.  To this end, we ask you to provide \textbf{rich
  fragmentation spectra} to SIRIUS, meaning that you \textbf{must not
  noise-filter} these spectra, or let the peak picking/centroiding software
  do that for you.  At present, SIRIUS considers up to 60 peaks in the
  fragmentation spectrum, and decides for itself which of these peaks are
  considered noise.

\item You will find that CSI:FingerID can sometimes identify the correct
  structure although the fragmentation spectrum is (almost) empty --- do not
  get fooled, this is often nothing but lucky guessing.  If you know how to
  structurally elucidate a compound based on an empty spectrum, please
  contact us and tell us how.

\item You may have heard that peaks in a MS/MS spectrum with high mass
  carry more information than peaks with low mass: This is a
  misunderstanding.  For example, if CSI:FingerID has to differentiate
  between 10000 candidates with identical molecular formula, then observing a
  fragment corresponding to an \ce{H2O} loss is in fact very uninformative.
  To this end, \textbf{do not set up your instrument to favor peaks of large
  masses}, sacrificing those with smaller masses.

\item Some instrument types (e.g., time-of-flight) suffer from detectors that
  can run into saturation; saturated peaks can have mass differences much
  larger than those expected for other peaks.  Unfortunately, most peak
  picking software do not mark such peaks as ``misshaped''.  To this end, it
  is possible that the most intense peak in a spectrum is not explained, as
  its mass deviation is extremely high.
\end{itemize}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Monoisotopic masses}

The monoisotopic mass of a molecule (or ion) is formally defined as ``the sum
of masses of the atoms in a molecule (or ion) using the unbound,
ground-state, rest mass of the most abundant isotope for each element.''
Using this definition, the monoisotopic mass is usually not the most abundant
isotopologue of the molecule (e.g., peptides and proteins), it is often not
resolved from other isotopologue peaks, and it may be undetectable in an
MS experiment as it has intensity below noise level.  In particular, given
the isotope pattern of an unknown molecule, it is generally impossible to
determine which of the peaks correspond to the monoisotopic peak.  In total,
this definition is not very practical.

Many researchers that work on the simulation and interpretation of isotope
patterns have therefor introduced a slightly different and more practical
definition of the monoisotopic mass of a molecule, see for
example \citet{dittwald15fine, meusel16predicting}: Here, the isotopologue of
a molecule where each atom is the isotope with the lowest nominal mass
(according to the natural isotope distribution of elements) is referred to
as \emph{monoisotopic}.  This definition has the advantages that the
monoisotopic mass of a molecule is always the sum of monoisotopic masses of
the atoms, which can be defined analogously; the monoisotopic peak is in all
cases the first peak of the ideal isotope pattern; and, the monoisotopic
(isotopologue) peak is always resolved from all other isotopologue peaks,
even at unit mass accuracy.  Clearly, the monoisotopic peak of a molecule may
again be undetectable in an MS experiments.

SIRIUS uses the second, more practical definition of ``monoisotopic''.  This
results in notable differences only for molecules that contain ``uncommon
elements'' such as boron or selenium.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Theoretical masses of ions}

There are different ways of computing the mass of an ionized molecule such as
\ce{C6H7O+} or \ce{C6H6ONa+} that will result in slightly different results: in
particular, adding the mass of a proton vs.\ subtracting the mass of an
electron.  Following suggestions by \citet{ferrer07importance}, SIRIUS
computes this mass by \emph{subtracting the rest mass of an electron}.  To
this end, the monoisotopic mass of \ce{C6H7O+} is the monoisotopic mass of
the molecule \ce{C6H7O} (95.049690~Da) minus the rest mass of an electron
(0.000549~Da), which totals as 95.049141~Da.  Similarly, the monoisotopic
mass of \ce{C6H6ONa+} equals 117.031634~Da - 0.000549~Da = 117.031085~Da.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{table}
\caption{Isotopes with masses and abundances as used by SIRIUS.  In this
table, \emph{masses have been rounded to six decimals} for the purpose of
presentation; internally, SIRIUS uses masses with higher precision.  `AN'
is atomic number.  *Isotope abundances of boron can vary strongly, so isotope
pattern analysis is of little use for identifying the correct molecular
formula in case boron is present.}
\label{tab:isotope-masses}
\smallskip

\begin{center}
\begin{tabular}{rr|c >{$}r<{\,\%$} >{$}r<{$}}
element (symbol) & AN & isotope & \multicolumn{1}{r}{abundance} & \text{mass
(Da)} \\

\hline

hydrogen (\ce{H}) & 1 & \ce{^{1}H} & 99.988 & 1.007825 \\
 & & \ce{^{2}H} & 0.012 & 2.014102 \\[0.5ex]

boron (\ce{B}) & 5
  & \ce{^{10}B} & 19.9^* & 10.012937 \\
 && \ce{^{11}B} & 80.1^* & 11.009305 \\[0.5ex]

carbon (\ce{C}) & 6 & \ce{^{12}C} & 98.93 & 12.0 \\
 & & \ce{^{13}C} & 1.07 & 13.003355 \\[0.5ex]

nitrogen (\ce{N}) & 7 & \ce{^{14}N} & 99.636 & 14.003074 \\
 & & \ce{^{15}N} & 0.364 & 15.001090 \\[0.5ex]

oxygen (\ce{O}) & 8 & \ce{^{16}O} & 99.757 & 15.994915 \\
 & & \ce{^{17}O} & 0.038 & 16.999131 \\
 & & \ce{^{18}O} & 0.205 & 17.999160 \\[0.5ex]

fluorine (\ce{F}) & 9 & \ce{^{18}F} & 100 & 18.000938 \\[0.5ex]

silicon (\ce{Si}) & 14
  & \ce{^{28}Si} & 92.223 & 27.976927 \\
 && \ce{^{29}Si} &  4.685 & 28.976495 \\
 && \ce{^{30}Si} &  3.092 & 29.973770
 \\[0.5ex]

phosphor (\ce{P}) & 15 & \ce{^{31}P} & 100 & 30.973762 \\[0.5ex]

sulfur (\ce{S}) & 16 & \ce{^{32}S} & 94.99 & 31.972071 \\
 & & \ce{^{33}S} & 0.75 & 32.971459 \\
 & & \ce{^{34}S} & 4.25 & 33.967867 \\
 & & \ce{^{36}S} & 0.01 & 35.967081 \\[0.5ex]

chlorine (\ce{Cl}) & 17
  & \ce{^{35}Cl} & 75.76 & 34.968853 \\
 && \ce{^{37}Cl} & 24.24 & 36.965903
 \\[0.5ex]

selenium (\ce{Se}) & 34
  & \ce{^{74}Se} &  0.89 & 73.922476 \\
 && \ce{^{76}Se} &  9.37 & 75.919214 \\
 && \ce{^{77}Se} &  7.63 & 76.919914 \\
 && \ce{^{78}Se} & 23.77 & 77.917309 \\
 && \ce{^{80}Se} & 49.61 & 79.916521 \\
 && \ce{^{82}Se} &  8.73 & 81.916699
 \\[0.5ex]

bromine (\ce{Br}) & 35
  & \ce{^{79}Br} & 50.69 & 78.918337 \\
 && \ce{^{81}Br} & 49.31 & 80.916291 \\[0.5ex]

iodine (\ce{I}) & 53
 & \ce{^{127}I} & 100 & 126.904473 \\[0.5ex]
\end{tabular}
\end{center}
\end{table}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Above, masses have been rounded to six decimals; internally, SIRIUS uses
double precision
\footnote{\sphinxAtStartFootnote\url{https://en.wikipedia.org/wiki/Double-precision_floating-point_format}}
for representing masses.  Masses of isotopes are taken from the AME2016
atomic mass evaluation \todo{passt das jetzt?}  \citep{wang17ame2016}.  See
Table~\ref{tab:isotope-masses} for the isotope masses and abundances used by
SIRIUS, again rounded to six decimals for presentation.

\textbf{We suggest to calibrate your instrument with ion masses as calculated
above.}  In any case, you should \emph{be aware of this tiny mass
difference}, as this can result in unexpected behavior when decomposing
masses; see for example Pluskal \emph{et al.}, \emph{Anal Chem} 84:4396--4403.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Mass deviations}
\label{sec:mass-deviations}

SIRIUS assumes that mass deviations (the difference between the measured
mass and the theoretical mass of the ion) are normally distributed
\citep{jaitly06robust, zubarev07proper, boecker16fragmentation}.  The
user-defined parameter ``mass accuracy'' is given in parts-per-million (ppm).
SIRIUS interpretes this parameter as a ``guarantee'' and, hence, assumes
that \textbf{this is the maximum allowed mass deviation}; it
will \textbf{discard} all explanations that require a larger mass deviation.
This implies that \textbf{if in doubt, you should use a larger mass accuracy}
to ensure that SIRIUS can successfully annotate peaks in the spectrum.  
%By
%default, SIRIUS further assumes that this mass accuracy is 3 times the
%standard deviation $\sigma$ of the normal distribution, and mass deviations
%are distributed according to $\mathcal{N}(0,\sigma^2)$.  Under this
%assumption $99.7\,\%$ of the mass deviations fall within the specified range,
%whereas larger mass deviations are truncated.  
For masses below 200~Da, we
use the absolute mass deviation at 200~Da, as we found that small masses vary
according to an absolute rather than a relative error.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Molecular formulas}

Unless instructed otherwise, SIRIUS will consider all molecular formulas
that are chemically feasible and explain the precursor mass of the
molecule/ion: For example, if your query compound is pinensin~A
(\ce{C96H139N27O30S2}, monoisotopic mass $2213.962$~Da) then SIRIUS will
consider all 19\,746\,670 candidate molecular formulas that explain this
monoisotopic mass (assuming set of elements \ce{CHNOPS}, see below, and
10~ppm mass accuracy).  SIRIUS penalizes candidate molecular formulas that
deviate too strongly of what we assume a molecular formula of a biomolecule
to look like (for example, \ce{C2H2N12O12} will receive a penalty), but this
penalty is used cautiously: Only 2.6\,\% of the molecular formulas of all
PubChem compounds --- and, hence, only a tiny fraction of molecular formulas
from compounds not marked as biomolecules --- are penalized.  Molecular
formulas are never rewarded by SIRIUS.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\begin{table}
%\caption{\todo{Outlier molecular formulas not penalized by SIRIUS}, despite
%being classified as ``not biomolecule-like''.}
%\label{tab:outlier-mfs}
%\smallskip
%
%\begin{center}
%\begin{tabular}{rrrr}
%\ce{H2O} & \ce{CO2} & \ce{C10HF19O2} & \ce{C10HF19O2} \\
%\ce{H2O} & \ce{CO2} & \ce{C10HF19O2} & \ce{C10HF19O2}
%\end{tabular}
%\end{center}
%\end{table}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

SIRIUS uses a short list of outlier molecular formulas which would be
penalized by the above method, as they are not ``biomolecule-like''; these
molecular formulas are not penalized, as they have been observed in
metabolomics experiments (for example, as solvents), but are also not
rewarded. These outlier molecular formulas
will be considered as candidates by SIRIUS even if they violate elemental
constrains such as ``at most 2 fluorine''.  
%See Table~\ref{tab:outlier-mfs}
%for a complete list of all molecular formulas which are explicitly not
%penalized.

Considering all molecular formulas implies that a set of elements has to be
provided from which these molecular formulas are generated.  SIRIUS
includes methods for the auto-detection of elements from the isotope and
fragmentation pattern of the query compound \citep{meusel16predicting}.

In case you compound is large (above 600~Da) or you have incomplete
information (no isotope pattern), you can restrict SIRIUS to only consider
molecular formulas found in PubChem.  Doing so, it is not possible to ever
detect molecules with a novel molecular formula, though.

Recent evaluations (for example, as part of the CASMI
contest\footnote{\sphinxAtStartFootnote\url{http://casmi-contest.org/}})
indicated that one can determine molecular formulas by searching in a
structure database using tool such as MAGMa, CFM-ID or CSI:FingerID.
(Somewhat consequently, the CASMI~2016 contest did no longer have a category
for molecular formula identification.)  \textbf{We strongly advice against
doing so, as this is apparently a hidden prior problem.}  Molecular formulas
in a structure database are far from being uniformly distributed: For a
certain precursor mass plus mass inaccuracy, you may find 90\% molecular
structures with only one molecular formulas.  Assuming that the molecular
structures in our evaluation set are uniformly chosen at random, even a
method that uniformly draws a molecular structure, then reports its molecular
formula will get 81\% correct identifications for the molecular formula
identification task.  A method that still ignores the mass spectrometry data
beyond the monoisotopic mass, but reports the majority vote in the structure
database would even reach 90\% correct identifications.  (This is comparable
to an app for bird identification that allways outputs ``It is a house
sparrow!'' for Britain, because the house sparrow is the most common bird in
Britain.  This app will get a lot of correct identifications.)  Clearly,
computational methods such as MAGMa, CFM-ID or CSI:FingerID are not random,
but they will fall for this pit, too.  To this end, we advice for the
``classical chemical identification pipeline'' where \textbf{molecular
formulas are identified first, then molecular structure.}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Fragmentation trees}

Fragmentation trees annotate the fragmentation spectrum with molecular
formulas, and identify likely losses between the ions in the fragmentation
spectrum.  Fragmentation trees can be used both to identify the molecular
formula of a query compound, and to derive information about its
fragmentation: For example, this is used in CSI:FingerID to predict the
molecular fingerprint of the query compound.  Fragmentation trees are
computed directly from the fragmentation spectrum, and do not use or require
any spectral libraries or molecular structure databases (for the subtle
``exemptions'' from this rule, see \citet{boecker16fragmentation}).
Fragmentation trees are computed by combinatorial optimization; the
underlying optimization problem constitutes a Maximum Aposterior Estimator.
The optimization problem (finding a maximum colorful subtree) is NP-hard but
nevertheless solved optimally, explaining why computations sometimes require
significant running time for large molecules with rich fragmentation spectra.

If you think that the computation process should be further speeded up, we
ask you to cite our papers on swiftly computing fragmentation trees
\citep{white15speedy, rauf13finding}, as this would give us an incentive to
continue our work on this topic: We stress that the current version of
SIRIUS is \textbf{many million times faster} than the initial version
\citep{boecker08towards}.  In fact, this initial version could not process
more than 15 peaks in the fragmentation spectrum, due to exploding running
times \emph{and} memory requirements.

Modeling the fragmentation process as a tree comes with two restrictions:
Namely, ``pull-ups'' and ``parallelograms''.  A \textbf{pull-up} is a
fragment which is inserted too deep into the trees.  Due to our combinatorial
optimization, SIRIUS will try to generate deep trees, assuming that there
are many small fragmentation steps instead of few larger ones.  SIRIUS
will, for example, prefer three consecutive \ce{C2H2} losses to a
single \ce{C6H6} loss.  This does not affect the quality of the molecular
formula identification; but when interpreting fragmentation trees, you should
keep in mind this side effect of the combinatorial
optimization.  \textbf{Parallelograms} are consecutive fragmentation
processes that happen in more than one order: For example, the precursor ion
looses \ce{H2O} then \ce{CO2}, \emph{but also} \ce{CO2} then \ce{H2O}.
SIRIUS will always decide for one order of such fragmentation reactions, as
this is the only valid way to model the fragmentation as a tree.

At present, SIRIUS does not make use of isotope patterns in the
fragmentation spectra: Usually, SIRIUS will simply flag these peaks and
ignore them in the optimization process.  This will likely change in an
upcomming version of SIRIUS.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Molecular fingerprints}

\emph{Molecular fingerprints} can be used to encode the structure of a
molecule: Most commonly, these are binary vector of fixed length where each
bit describes the presence or absence of a particular, fixed \emph{molecular
property}, usually the existence of a certain substructure.  As an example,
consider PubChem CACTVS fingerprints with length 881 bits: Molecular property
121 encodes the presence of at least one ``unsaturated non-aromatic
heteroatom-containing ring size~3''.  Most bits are just explained via their
SMARTS (SMiles ARbitrary Target Specification)
string\footnote{\sphinxAtStartFootnote%
\url{https://en.wikipedia.org/wiki/Smiles_arbitrary_target_specification}}:
For example, molecular property 357 of PubChem CACTVS encodes SMARTS string
``\verb=[#6](~[#6])(:c)(:n)='', corresponding to a central carbon atom
connected to a second carbon atom via any bond, to a third aromatic carbon
atom via an aromatic bond, and to an aromatic nitrogen atom via an aromatic
bond.
See \url{ftp://ftp.ncbi.nlm.nih.gov/pubchem/specifications/pubchem_fingerprints.pdf}
for the full description of the CACTVS fingerprint. We ignore all molecular
properties that can be derived from the molecular formula of the query
compound (for example, bits 0 to 114 of PubChem CACTVS).

Given the molecular structure of a compound, we can deterministically compute
its molecular fingerprint: We use the Chemical Development Kit CDK
\citep{steinbeck03chemistry, steinbeck06recent, willighagen17chemistry} for
this purpose.  \citet{heinonen12metabolite} pioneered the idea of predicting
a complete molecular fingerprint from the fragmentation spectrum of a query
compound: Before this, only few, usually hand-selected properties (presence
or absence of certain substructures) were predicted from fragmentation
spectra, in particular for GC-MS with Electron Ionization;
see \citet{curry90msnet} for an excellent example.

Given the fragmentation spectrum and fragmentation tree of a query compound,
CSI:FingerID predicts its molecular fingerprint using Machine Learning
(linear Support Vector Machines), see \citet{shen14metabolite}
and \citet{duehrkop15searching} for the technical details.  CSI:FingerID does
not predict a single fingerprint type but instead, five of them: Namely, CDK
Substructure fingerprints, PubChem CACTVS fingerprints, Klekota-Roth
fingerprints~\citep{klekota08chemical}, FP3 fingerprints, and MACCS
fingerprints.  In addition, CSI:FingerID predicts ECFP2 and ECFP4 fingerprints
\citep{rogers10extended} that appear sufficiently often in the training data.
Different from other fingerprints, ECFP are not encoded via SMARTS matching;
instead, a hash function encodes the neighborhood of each atom in the
molecule.  In principle, these fingerprints can encode $2^{32} \approx
4.2 \cdot 10^9$ different substructures (molecular properties); in practice,
it is possible but very unlikely that two substructures share the same value,
due to a hash collision.

CSI:FingerID predicts only those molecular properties that showed reasonable
prediction quality in cross validation (F$_1$ at least $0.25$, see below).
In total, $3\,215$ molecular properties are predicted by CSI:FingerID~1.1.

CSI:FingerID does not only predict if some molecular property is zero
(absent) or one (present); it also provides an \textbf{estimate how sure it
is about this prediction}.  Mathematically speaking, we estimate the
posterior probability that the molecular property is present: Estimates close
to one indicate that CSI:FingerID is rather sure that the molecular property
is present; similarly, estimates close to zero for an absent molecular
property; whereas estimates between $0.1$ and $0.9$ hint towards an unsure
situation.  Posterior probabilities are estimated using a method
by \citet{platt00probabilistic}, so we also refer to these estimates as
``Platt probabilities''.  \textbf{But even if CSI:FingerID is 99\,\% sure
that a molecular property is present, this does not mean that it is indeed
present!}  CSI:FingerID predicts thousands of molecular properties, and 10
out of 1000 predictions should be incorrect at this level of accuracy.
Furthermore, estimation parameters were derived from the training data, and
if your query molecule structures are very different from those in the
training data, it is rather likely that some estimates are imprecise.  In
addition to Platt probabilities, we also report the performance of each
molecular property classifier in cross validation: The F$_1$
score\footnote{\url{https://en.wikipedia.org/wiki/F1_score}} is the harmonic
mean\footnote{\url{https://en.wikipedia.org/wiki/Harmonic_mean}} of precision
(fraction of retrieved instances that are relevant) and recall (fraction of
relevant instances that are
retrieved).\footnote{\url{https://en.wikipedia.org/wiki/Precision_and_recall}}
Molecular properties that have a classifier with F$_1$ score close to one,
are more trustworthy than those with F$_1$ score close to zero; again, this
has to be treated with some care, as these measures were estimated from the
training data using cross validation.

It is important to understand that the predicted molecular fingerprint which
is returned by the CSI:FingerID web service, has \emph{per~se} no connections
to any structures in any molecular structure database.  That means
that \textbf{even if the correct molecular structure is not contained in any
structure database, the predicted fingerprint is still valid} within the
prediction power of the method.  For example, you can use it to hypothesize
about the structure of an ``unknown unknown'' not present in any structure
database.  We have added a tab in the Graphical User Interface that allows
you to examine the predicted molecular fingerprint.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Molecular structures}

By default, SIRIUS searches in either PubChem or a biomolecule structure
database; in addition, SIRIUS now offers to search in your own ``suspect
database''.
%
\begin{itemize}
\item When searching \emph{PubChem}, we use a local copy of the database
  where we have precomputed all molecular fingerprints, as computing the
  fingerprints of the candidates ``on the fly'' is too time-consuming.  We
  are sporadically updating our local copy of PubChem.  
  You can lookup the date of the latest database update in the database dialog.
  %\todo{There should be
  %a way for the user to find out what PubChem copy was used; same for bio-db.
  %How about a web server command which returns a flat text file describing DB
  %statistics?  something like ``PubChem (22-09-2016) 894567 structures RETURN
  %KNApSAcK (23-03-2017) 1234 structures etc''}
  % we report this in the database dialog.

\item The \emph{biomolecule structure database} is an amalgamation of several
  structure databases that contain biomolecules (metabolites and other
  compounds of biological relevance; molecules that are products of nature,
  or synthetic products with potential bioactivity).  Currently, this
  biomolecule structure database consists of KNApSAcK \citep{shinbo06plant},
  HMDB \citep{wishart13hmdb}, ChEBI \citep{hastings13chebi},
  KEGG \citep{kanehisa16kegg}, HSDB \citep{fonger14national},
  MaConDa \citep{weber12maconda}, BioCyc \citep{caspi14metacyc},
  UNPD \citep{gu03unpd}, a subset of biomolecules from
  ZINC \citep{irwin12zinc}, all structures from GNPS \citep{wang16sharing}
  and MassBank \citep{horai10massbank}, and MeSH-annotated compounds from
  PubChem \citep{nelson01relationships, kim16pubchem}.
\end{itemize}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Training data}

The fragmentation tree computation of SIRIUS \textbf{is not trained on any
data}, since no machine learning is used for this step.  The parameters for
fragmentation tree computation were estimated from two MS/MS spectra
datasets, with 2005 compounds from GNPS \citep{wang16sharing} and 2046
compounds from Agilent (``MassHunter Forensics/Toxicology PCDL'' version
B.04.01 from Agilent Technologies Inc., Santa Clara, CA, USA).  Parameters of
this step were not optimized to maximize, say, the molecular formula
identification rate, and estimates should be very robust.  All spectra were
recorded in positive ion mode.  Fragmentation tree computation and molecular
formula estimation appear to work very well for negative ion mode data, too;
but there is no guarantee for that.

The Machine Learning part of CSI:FingerID, namely the essemble of linear
Support Vector Machines, is currently (CSI:FingerID version~1.1) trained
on $12\,108$ compounds from \citep{wang16sharing} and $2\,073$ compounds
from MassBank \citep{horai10massbank}.  In total, CSI:FingerID is trained on
$14\,181$ compounds with $8\,210$ unique structures. Again, all spectra were
recorded in
positive ion mode.  Different from fragmentation tree computation, we assume
that the \textbf{prediction of molecular fingerprints is challenging for
negative ion mode data}, given that we have no training data.  Hopefully,
enough negative ion mode data will become available in the near future, so
that we can retrain CSI:FingerID on this data and solve this issue for good.

\textbf{We would like to explicitly and emphatically thank everyone who made
their spectra publically available.}  With that, you have done a huge favor
not only to us, but to everyone in the metabolomics community; which,
unfortunately, is not recognized by the community at the moment.  We
sincerely hope that the metabolomics community will become aware of the
urgent need for open data and data sharing in the near future (just like the
genomics community did 25 years ago, or the proteomics community 10 year
ago); and that you will receive your well-deserved accolades then.
 
We are constantly adding new training data that becomes publically available.
If you have data from reference compounds, we ask you to upload these to a
public database such as GNPS or MassBank; if this is not possible for some
reason, \textbf{you can contact us so that we can add your data to the
CSI:FingerID training data without making it publically available}.  Please
help us improve the performance of CSI:FingerID by providing additional
training data!


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%  CHAPTER  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\chapter{Graphical User Interface}
\label{gui::doc}\label{gui:user-interface}

\begin{center}
  \includegraphics[width=0.8\textwidth]{{gui}.png}
\end{center}


\section{Overview}
\label{gui:overview}

Starting with version~3.1, our software ships with a Graphical User
Interface.  On top of the screen you find the toolbar~(1).  On the left side
is the compound list (2) displaying all imported compounds.
Each \gui{compound} lists MS and MS/MS spectra corresponding to a single
measured compound.  If a compound has been processed successfully, you will
see a tick mark on the right~(3); if something goes wrong during computation
you will see a cross symbol~(4).  The output of a computation is an ordered
list of suggested molecular formula candidates.  After selecting a compound
an overview is displayed.  It shows a list of all molecular formula
candidates~(5), sorted by score, the corresponding spectrum (6) and the
fragmentation tree of the selected candidate molecular formula~(7).
Explained peaks are highlighted in the spectrum.  Nodes in the fragmentation
tree are colored according to their score.  In the upper right corner are
settings and bug report dialogs~(8).
\todo{filter options etc}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Data import}
\label{gui:data-import}

SIRIUS offers two modes for data import: \gui{Single import} and \gui{Batch
import}. The single import is triggered when clicking on the \gui{Import}
button in the toolbar. It allows you to import \textbf{one} compound.  (We
will use the term ``compound'' as a description of MS and MS/MS spectra
belonging to a single compound.)  The single import mode is recommended if
your data consists of several CSV (comma separated values) files, such as the
data from the CASMI challenges. First press on \gui{Import} to start the
import dialog.

\begin{center}
  \includegraphics[width=0.8\textwidth]{{load}.png}
\end{center}

For each spectrum you have to select the MS level (either MS1 or MS/MS). If you have MSn spectra you can just import them as MS/MS spectra. You can select a name for the compound as well as an ionization mode. The collision energy is an optional attribute as it does not affect the computation.

You can import \code{.ms} and \code{.mgf} files using the \gui{Batch Import}. In this mode SIRIUS will read all attributes (MS level, ionization, parent mass) directly from the file. You can, however, change these attributes afterward by selecting the imported compound and clicking on the \gui{Edit} button.

See section \emph{Supported Input Formats} for a description of the file formats \code{.ms} and \code{.mgf}.


\subsection{Drag and drop}
\label{gui:drag-and-drop}
SIRIUS supports Drag and Drop: Just move your input files into the application window. This is usually the easiest way to import data into SIRIUS. Supported file formats for Drag and Drop are \code{.csv}, \code{.ms}, \code{.sirius} and \code{.mgf}.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Identifying molecular formulas with SIRIUS}
\label{gui:identifying-molecular-formulas-with-sirius}

As for importing data SIRIUS offers two computation modes: \gui{Single 
Computation} and \gui{Batch Computation}. The Single Computation allows you 
to setup different parameters for each compound. You can trigger it by 
right-clicking on a compound and choosing \gui{Compute} in the context 
menu. The Batch Computation will compute all compounds in the workspace.
Besides, you can select multiple compounds and choose \gui{Compute} to 
only compute a subset of your imported compounds.

\begin{center}
  \centering
\end{center}


\subsection{Parent mass}
\label{gui:parent-mass}

The exact m/z of the parent peak. If MS1 data is present, the m/z of the monoisotopic peak is presented as default. Otherwise, an autocompletion offers a list of high intensive peaks from the MS/MS spectra.


\subsection{Elements besides CHNOPS}
\label{gui:elements-besides-chnops}
SIRIUS will use the elements carbon (C), hydrogen (H), nitrogen (N), oxygen (O), phosphorus (P) and sulfur (S) by default. Additional elements can be selected within the \gui{Select elements} dialog. 
Adding additional elements will increase running time. Using (too many) elements that do not occur in the correct molecular formula of the compound might worsen the results.

The automated detection of a set of ``uncommon elements'' is available if the 
isotope pattern is provided. These elements are sulfur (S), chlorine (Cl), 
bromine (Br), boron (B), and selenium (Se). 
Using \gui{Auto detect} will clear your element selection and will set new values based on the detection. Autodetection is usually quite sensitive and rather overpredicts the actual quantity of an element.

\subsection{Other}
\label{gui:other}

The ionization mode determines the polarity of the measurement (positive or negative) as well as the adduct (e.g. protonation or sodium adduct). If you choose \gui{Unknown Positive} or \gui{Unknown Negative} SIRIUS will not care about the adduct, but report the molecular formula of the \textbf{ion} in the candidate list. Otherwise, SIRIUS will subtract the adducts formula from the ions formula and report neutral molecular formulas in the candidate list as well as in the fragmentation trees.

Choose either \gui{Q-TOF}, \gui{Orbitrap} or \gui{FT-ICR} in the instrument field. The chosen instrument affects only very few parameters of the method (mainly the allowed mass deviation). If your instrument is not one of these three then just select the Q-TOF instrument.

You can change the maximal allowed mass deviation in the \gui{ppm}
field. SIRIUS will only consider molecular formulas which mass deviations
below the chosen ppm; for masses below 200~Da, the allowed mass deviation
is \(200 \cdot \frac{ppm_{max}}{10^6}\).

Finally, you can select the number of molecular formula candidates that
should be reported in the output, and what molecular formulas are considered
as candidates: If you select option \gui{all possible molecular
formulas} then SIRIUS will enumerate over all molecular formulas that
match the ion mass, filtering out only molecular formulas with negative ring
double bond equivalent. If you choose \gui{all PubChem formulas} then
SIRIUS will select all molecular formulas from PubChem.  Option
\gui{organic PubChem formulas} ignores molecular formulas containing
elements untypical for organic compounds such as Si or Mg; molecular formulas
pass this filter if they are composed solely of \ce{CHNOPSBBrClIF}.  When
choosing \gui{formulas from biomolecule databases}, \todo{change name of
option in program, biomolecule instead of biological!} SIRIUS will use all
formulas contained in databases with biological compounds or compounds that
could be expected in biological experiments (e.g. KEGG, BioCyc, HMDB, but
also MaConDa).
%
\begin{description}
\item[{Please consider the following:}] \leavevmode\begin{itemize}
\item {} 
We never search in these databases directly, but rather in our local database
copies. Although we regularly update our database, it may happen that some
new compound in, say, ChEBI is not already contained in our local copy.

\item {} 
When choosing a \gui{molecular fromulas from a database} option, SIRIUS
will ignore your element restrictions and instead allow all elements.

\item {} 
We do not recommend to restrict molecular formula searching to biomolecule
databases, but doing so significantly speeds up computations, as SIRIUS has
to consider significantly less molecular formulas and download significantly
smaller candidate structure lists.
\end{itemize}
\end{description}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Identifying molecular structure with CSI:FingerID}
\label{gui:identifying-molecular-structure-with-csi-fingerid}

After computing the fragmentation trees you can search these in a structure database. Again we provide a \gui{single mode} and a \gui{batch mode}. The single mode is available by clicking on the molecular formula of interest, then switching to the \gui{CSI:FingerID} tab and pressing on the \gui{Search online with CSI:FingerID} button.
The batch mode can be triggered by pressing on the \gui{CSI:FingerID} in the toolbar.

\begin{center}
  \includegraphics[width=0.4\textwidth]{{fingerid}.png}
\end{center}

When starting the CSI:FingerID search you are again asked to choose between
PubChem or biomolecule databases. This is mainly a performance issue because
you can filter your result lists afterwards by any database you want to. Our
biomolecule database is several magnitudes smaller than PubChem and
downloading and searching structure lists from biomolecule databases is
significantly faster. However, when searching in biomolecule databases you
might never see if there are structures with possibly much better score from
PubChem. Therefore, we recommend to search in PubChem and filter the result
list if you expect the result to be contained in biomolecule databases.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Visualization of the results}
\label{gui:visualization-of-the-results}

Each compound has an \gui{Overview} panel to display the most important
information. The candidate list contains the best candidate molecular
formulas ordered by score. 
\todo{all these new ionizations}
Molecular formulas are always written in neutral
form, except for compounds with unknown ionization mode. For the selected
molecular formula candidates the \gui{Spectra view} visualizes which peak is
assigned to a fragment.  The corresponding fragmentation tree is visualized
in the \gui{Tree view}.  Both views can be displayed in a separate panel to
have a more detailed look.  The \gui{CSI:FingerID} panel displays candidates
from structure prediction.

\subsection{Overview tab}
\label{gui:overview-view}

The \gui{Overview} tab displays the candidate list, spectrum and
fragmentation tree of the selected candidate. Candidates are ordered by total
score, but can be sorted by any other column. Moreover, the list can be
filtered using the corresponding text field. A green row highlights the
molecular formula of the best candidate structure found by CSI:FingerID.

\begin{center}
  \includegraphics[width=0.8\textwidth]{{overview}.png}
\end{center}

The length of the bars for the different score columns (isotope pattern,
fragmentation pattern, total) as well as the displayed numbers for
columns \gui{Isotope Score} and \gui{Tree Score}, correspond
to \emph{logarithms} of maximum likelihoods (probability that this
hypothesis, i.e.\ molecular formula, will generate the observed data).  In
contrast, the number in the \gui{Score} column is the posterior probability
of the hypothesis (molecular formula), and these probablities sum to one.  A
higher posterior probability of the top hit may indicate that this molecular
formula has a higher chance of being correct; but we stress that \textbf{a
posterior probability of 90\,\%, must not be misunderstood as a 90\,\%
probability that this molecular formula identification is correct!}  The
displayed probabilities are neither q-values nor Posterior Error
Probabilities.


\subsection{Tree view tab}
\label{gui:tree-view}

\begin{center}
  \includegraphics[width=0.8\textwidth]{{tree_view}.png}
\end{center}

The \gui{Tree view} tab displays the estimated fragmentation tree. Each node
in this tree assigns a molecular formula to a peak in the (merged) MS/MS
spectrum. Each edge is a hypothetical fragmentation reaction. The user has
the choice between different node styles and color schemes.

The displayed fragmentation tree can be exported as JPEG, GIF, and
PNG. Alternatively, the Dot file format contains a text description of the
tree. It can be used to render the tree externally. The command-line tool
Graphviz can transform dot files into image formats (PDF, SVG, PNG etc). The
JSON format yields a machine-readable representation of the tree.


\subsection{Spectrum view tab}
\label{gui:spectrum-view}

\begin{center}
  \includegraphics[width=0.8\textwidth]{{spectra_view}.png}
\end{center}

In the \gui{Spectrum view} tab, all peaks that are annotated by the
fragmentation tree are colored in orange. Peaks that are annotated as noise
are colored black.  Hovering with the mouse over a peak shows its detailed
annotation.


\subsection{CSI:FingerID view}
\label{gui:csi-fingerid-view}

\begin{center}
  \includegraphics[width=0.8\textwidth]{{fingerprints}.png}
\end{center}

This tab shows you the candidate structures for the selected molecular formula ordered by the CSI:FingerID search score. If you want to filter the candidate list by a certain database (e.g. only compounds from KEGG and BioCyc) you can press the filter button. A menu will open displaying all available databases. Only candidates will be displayed that are enabled in this filter menu. Note that PubChem is enabled by default and, therefore, always the complete list is shown. If you want to see only compounds from KEGG and BioCyc you have to disable PubChem and enable KEGG and BioCyc.

Another way of filtering is the XLogP slider. If you have information about
retention times and expected logP values of your measured compound you can
use this slider to filter the candidate list by certain XLogP values. The
slider allows you to define min and max values. XLogP is calculated using the
Chemical Development Kit CDK \citep{steinbeck03chemistry, steinbeck06recent,
willighagen17chemistry}.

The blue and red squares are some visualization of the CSI:FingerID predictions 
and scoring. All blue squares represent molecular structures that are found in 
the candidate structure and are predicted by CSI:FingerID to be present in the 
measured compound. The more intense the color of the 
square the higher is the predicted probability for the presence of this 
substructure. The larger the square the more reliable is the predictor. 
The red squares, however, represent structures that 
are predicted to be absent but are, nevertheless, found in the candidate 
structure. Again, as more intense the square as higher the predicted 
probability 
that this structure should be absent. Therefore, a lot of large intense blue 
squares and as few as possible large intense red squares are a good indication 
for a correct prediction.

When hovering with the mouse over these squares the corresponding description of the molecular structure (usually a SMART expression) is displayed. When clicking on one of these squares, the corresponding atoms in the molecule that belong to this substructure are highlighted. If the substructure matches several times in the molecule, it is once highlighted in dark blue while all other matches are highlighted in a translucent blue.

Even if the correct structure is not found by CSI:FingerID --- in particular
if the correct structure is not contained in any database --- you can get
information about the structure by looking at the predicted structures: When
clicking on the large light green squares you see which molecular
substructures are expected in the measured compound.

You can open a context menu by right click on the compound. It offers you to open the compound in PubChem or copy the InChI or InChI key in your clipboard.

If the compound is contained in any biomolecule database, a blue label with the name of this database is displayed below the compound. You can click on most of these labels to open the database entry in your browser window.

You can export a single candidate list by clicking on the \gui{export list} button.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Workspace}
\label{gui:workspace}

All imported compounds together with their results (the candidate molecular formulas and corresponding fragmentation trees) form the \textbf{workspace}. You can save the workspace into a file by clicking on \gui{Save Workspace} in the toolbar. Analogously, you can load a workspace using \gui{Load Workspace}. The workspace is saved in the \code{.sirius} format, which is simply a Zip Archive containing the input spectra and fragmentation trees. You can share the \code{.sirius} files with other people. But be aware that the input spectra are contained in these files!

The SIRIUS Commandline Tool is able to output \code{.sirius} files by using the option \code{-O sirius}. You can import these files with the SIRIUS Graphical User Interface to get a visualization of the results.

Currently, CSI:FingerID predictions are not stored in the \code{.sirius} file. This will hopefully change in future releases.


\subsection{Export results}
\label{gui:export-results}

You can export your results as a simple \code{csv} file for all compounds by
clicking on the \gui{Export Results} button and selecting single file
export. Such a \code{.csv} file can then be imported
into \emph{Excel}, \emph{Matlab} or \emph{R}.
%
\begin{description}
\item[{The CSV file contains the following fields:}] \leavevmode\begin{itemize}
\item {} 
name of the compound

\item {} 
parent mass

\item {} 
ionization

\item {} 
for each molecular formula candidate there are two columns: one with the molecular formula and one with the corresponding score

\end{itemize}

\end{description}

If the number of molecular formula candidates differ between compounds, the number of fields per row might differ, too. However, most software programs do not have a problem with such \emph{malformed} CSV files.

Selecting the \gui{multiple files} option will create one file per compound. Exporting the \gui{Tree results} will give you a \code{.csv} file for each compound containing the molecular formula candidates with the following fields: 
\begin{itemize}
\item molecular formula
\item rank
\item overall score
\item fragmentation tree score
\item isotope pattern score
\item number of peaks which where annotated with a fragment molecular formula
\item ratio of explained intensity. It is the sum of intensities of all annotated peaks divided by the total intensity.
\end{itemize}
Exporting \gui{CSI:FingerID results} will create one \code{.csv} file for each compounds with the following fields:
\begin{itemize}
\item InChI key's first 14 positions (no stereochemistry information)
\item InChI
\item molecular formula
\item rank
\item score
\item name of the candidate compound (if available)
\item SMILES 
\item XlogP
\item PubChem identifier
\item occurrence in additional databases (if available)
\end{itemize}


\todo{Workspace and CLI output same format as .sirius?}


%\subsection{Confidence view}
%\label{gui:confidence-view}
%When identifying large amounts of spectra with CSI:FingerID you will, 
%unfortunately, get a lot of wrong predictions among the correct ones. With 
%We offer a confidence score that reflects the likelihood that some
%identification is correct. You can list all identifications by opening the 
%\gui{identification} tab.
%\begin{figure}[htb]
%\centering
%
%\includegraphics{{confidence}.png}
%\end{figure}
%
%This list contains the top hits in all structure candidate lists ordered by 
%their confidence. Entries at the top are more likely to be correct while 
%entries at the bottom are more likely to be wrong.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Settings}\label{sec:settings}
\todo{General, Proxy, Error report}
\begin{itemize}
\item \gui{General settings}
\begin{itemize}
	\item \gui{Allowed solvers:} chose the ILP solver for SIRIUS to use for fragmentation tree computation. GLPK is free, Gurobi is commericial but offers free academic license.
	\item \gui{Database cache:} location of cache directory. CSI:FingerID download candidate structures from our server and caches them for faster retrieval.
\end{itemize}

\item \gui{Proxy settings}
\begin{itemize}
	\item Sirius support three different kinds of proxy configuration SYSTEM, 
	SIRIUS and NONE. If SYSTEM (default) is select Sirius uses the system wide 
	Java proxy settings. If SIRIUS is selected it uses the configuration you 
	have specified int the Settings -> Proxy panel. If NONE is selected 
	Sirius ignores all proxy settings.
	  
	\item Edit the information in the Settings -> Proxy panel if you want to 
	address CSI:FingerID via a proxy server. Your specified configuration will 
	be tested if you hit the save button (see Figure below).
	
\begin{center}
  \includegraphics[width=0.4\textwidth]{{proxySetting}.png}
  \includegraphics[width=0.4\textwidth]{{connectionCheck}.png}
\end{center}
\end{itemize}

\item \gui{Error report settings}
\begin{itemize}
	\item Add an email address which will be sent with a bug report. This makes it possible for us to contact you, in case we need additional information to solve your problem. 
	\item Decide whether specific hardware and operating system information is send with your bug report.
\end{itemize}

\end{itemize}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Bug Reports}

We do our best so that you will not be confronted with errors while using
SIRIUS.  But we cannot test every possible scenario.  We encourage you to
send us a bug report in case you encounter an error.  It is very helpful if
you specify your email address. Often, errors are very specific and can only
be reproduced and understood with help of the input file and knowledge of the
used parameter settings. Therefore, we might reach out to you. \todo{or add
an upload field?}  With your help, we will continue to improve SIRIUS.  You
can also contact us at \url{sirius@uni-jena.de}.



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%  CHAPTER  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\chapter{SIRIUS Commandline Tool}
\label{commandline::doc}\label{commandline:sirius-commandline-tool}

\todo{what has changed in here? custom databases?}

The SIRIUS commandline tool can be either called via the binary by simply running the command \code{sirius} in your commandline. Alternatively, you can run the sirius jar file using Java with the command:

\begin{Verbatim}[commandchars=\\\{\}]
java \PYGZhy{}jar sirius.jar
\end{Verbatim}

You can always use the \code{-{-}help} option to get a documentation about the available commands and options. Assuming you want to analyze the example data given in the CASMI \footnote{\sphinxAtStartFootnote%
\url{http://casmi-contest.org/2014/example/MSpos\_Challenge0.txt}
} contest, you would execute the following on the commandline:
\begin{quote}

sirius -1 MSpos\_Challenge0.txt -2 MSMSpos\_Challenge0.txt
\end{quote}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Supported Input Formats}
\label{commandline:inputformats}\label{commandline:supported-input-formats}

\subsection{Mass spectra}
\label{commandline:mass-spectra}
The input of SIRIUS are MS and MS/MS spectra as simple peak lists. SIRIUS can read CSV files which contain on each line a m/z and an intensity value separated by either a whitespace, a comma or a TAB character. For example:

\begin{Verbatim}[commandchars=\\\{\}]
185.041199 4034.674316
203.052597 12382.624023
245.063171 50792.085938
275.073975 124088.046875
305.084106 441539.125
335.094238 4754.061035
347.09494 13674.210938
365.105103 55487.472656
\end{Verbatim}

The intensity values can be arbitrary floating point values. SIRIUS will transform the intensities into relative intensities, so only the ratio between the intensity values is important.

SIRIUS also supports the MGF (Mascot Generic Format). This file format was developed for peptide spectra for the mascot search engine. Each spectrum in a MGF file can contain many spectra each starting with \code{BEGIN IONS} and ending with \code{END IONS}. Peaks are again written as pairs of m/z and intensity values separated by whitespaces with one peak per line. Further meta information can be given as NAME=VALUE pairs. SIRIUS recognizes the following meta information:
\begin{itemize}
\item {} 
PEPMASS: contains the measured mass of the ion (e.g. the parent peak)

\item {} 
CHARGE: contains the charge of the ion. As SIRIUS supports only single charged ions, this value can be either 1+ or 1-.

\item {} 
MSLEVEL: should be 1 for MS spectra and 2 for MS/MS spectra. SIRIUS will treat higher values automatically as MS/MS spectra, although, it might be that it supports MSn spectra in future versions.

\end{itemize}

This is an example for a MGF file:

\begin{Verbatim}[commandchars=\\\{\}]
BEGIN IONS
PEPMASS=438.32382
CHARGE=1+
MSLEVEL=2
185.041199 4034.674316
203.052597 12382.624023
245.063171 50792.085938
275.073975 124088.046875
305.084106 441539.125
335.094238 4754.061035
347.09494 13674.210938
365.105103 55487.472656
END IONS
\end{Verbatim}

See also the GNPS \footnote{\sphinxAtStartFootnote%
\url{http://gnps.ucsd.edu/}
} database for other examples of MGF files.

A disadvantage of these data formats is that they do not contain all information necessary for SIRIUS to perform the computation. Missing meta information have to be provided via the commandline. Therefore, SIRIUS supports also an own file format very similar to the MGF format above. The file ending of this format is \code{.ms}. Each file contains one measured compound (but arbitrary many spectra). Each line may contain a peak (given as m/z and intensity separated by a whitespace), meta information (starting with the \textbf{\textgreater{}} symbol followed by the information type, a whitespace and the value) or comments (starting with the \textbf{\#} symbol). The following fields are recognized by SIRIUS:
\begin{itemize}
\item {} 
\textgreater{}compound: The name of the measured compound (or any placeholder). This field is \textbf{mandatory}.

\item {} 
\textgreater{}parentmass: the mass of the parent peak

\item {} 
\textgreater{}formula: The molecular formula of the compound. This
information is helpful if you already know the correct molecular formula and
just want to compute a fragmentation tree or recalibrate the spectrum

\item {} 
\textgreater{}ion: the ionization mode. See {\hyperref[commandline:ions]{\crossref{\DUrole{std,std-ref}{Ion Modes}}}} for the format of ion modes.

\item {} 
\textgreater{}charge: is redundant if you already provided the ion mode. Otherwise, it gives the charge of the ion (1 or -1).

\item {} 
\textgreater{}ms1: All peaks after this line are interpreted as MS peaks

\item {} 
\textgreater{}ms2: All peaks after this line are interpreted as MS/MS peaks

\item {} 
\textgreater{}collision: The same as \textgreater{}ms2 with the difference that you can provide a collision energy

\end{itemize}

An example for a .ms file:

\begin{Verbatim}[commandchars=\\\{\}]
\PYGZgt{}compound Gentiobiose
\PYGZgt{}formula C12H22O11
\PYGZgt{}ionization [M+Na]+
\PYGZgt{}parentmass 365.10544

\PYGZgt{}ms1
365.10543 85.63
366.10887 11.69
367.11041 2.67

\PYGZgt{}collision 20
185.041199 4034.674316
203.052597 12382.624023
245.063171 50792.085938
275.073975 124088.046875
305.084106 441539.125
335.094238 4754.061035
347.09494 13674.210938
365.105103 55487.472656
\end{Verbatim}


\subsubsection{Ion modes}
\label{commandline:ions}\label{commandline:ion-modes}
Whenever SIRIUS requires the ion mode, it should be given in the following format:

\begin{Verbatim}[commandchars=\\\{\}]
[M+ADDUCT]+ for positive ions
[M+ADDUCT]\PYGZhy{} for negative ions
[M\PYGZhy{}ADDUCT]\PYGZhy{} for losses
[M]+ for instrinsically charged compounds
\end{Verbatim}

ADDUCT is the molecular formula of the adduct. The most common ionization modes are \code{{[}M+H{]}+}, \code{{[}M+Na{]}+}, \code{{[}M-H{]}-}, \code{{[}M+Cl{]}-}. Currently, SIRIUS supports only single-charged compounds, so \code{{[}M+2H{]}2+} is not valid. For intrinsic charged compounds \code{{[}M{]}+} and \code{{[}M{]}-} should be used.


\subsubsection{Molecular formulas}
\label{commandline:molecular-formulas}\label{commandline:formulas}

Molecular Formulas in SIRIUS must not contain brackets.
Hence, \code{2(C2H2)} is not a valid molecular formula; write \code{C4H4}
instead.  Furthermore, all molecular formulas in SIRIUS are always neutral,
and there is no possibility to add a charge on a molecular formula (instead,
charges are given separately). Hence, \code{CH3+} is not a valid molecular
formula.  Write \code{CH3} instead, and provide the charge separately via
commandline option.


\subsubsection{Chemical alphabets}
\label{commandline:alphabets}\label{commandline:chemical-alphabets}
Whenever SIRIUS requires the chemical alphabet, you have to provide which elements should be considered and what is the maximum amount for each element. Chemical alphabets are written like molecular formulas. The maximum amount of an element is written in square brackets behind the element. If no square brackets are given, the element might occur arbitrary often. The standard alphabet is CHNOP{[}5{]}S, allowing the elements C, H, N O and S as well as up to five times the element P.


\paragraph{Identifying Molecular Formulas}
\label{commandline:identifying-molecular-formulas}
The main purpose of SIRIUS is identifying the molecular formula of the measured ion. The syntax for this command is:

\begin{Verbatim}[commandchars=\\\{\}]
sirius [OPTIONS] \PYGZhy{}z \PYGZlt{}PARENTMASS\PYGZgt{} \PYGZhy{}i \PYGZlt{}IONIZATION\PYGZgt{} \PYGZhy{}1 \PYGZlt{}MS FILE\PYGZgt{} \PYGZhy{}2 \PYGZlt{}MS/MS FILE\PYGZgt{}
\end{Verbatim}

Where MS FILE and MS/MS FILE are either CSV or MGF files. If MGF files are used, you might omit the PARENTMASS option. If you omit the IONIZATION option, {[}M+H{]}+ is used as default. It is also possible to give a list of MS/MS files if you have several measurements of the same compound with different collision energies. SIRIUS will merge these MS/MS spectra into one spectrum.

If your input files are in \code{.ms} or \code{.mgf} format (containing MSLEVEL and PEPMASS meta information), you can omit the -1 and -2 flag. For example:

\begin{Verbatim}[commandchars=\\\{\}]
sirius [OPTIONS] demo\PYGZhy{}data/ms
\end{Verbatim}

SIRIUS will pick the meta information (parentmass, ionization etc.) from the \code{.ms} files in the given directory. This allows SIRIUS to run in batch mode (analyzing multiple compounds without starting a new jvm process every time).

SIRIUS will output a candidate list containing the \gui{rank}, \gui{overall score}, \gui{fragmentation pattern score}, \gui{isotope pattern score}, the number of \gui{explained peaks} and the relative amount of \gui{explained intensity}. See the following example output:

\begin{Verbatim}[commandchars=\\\{\}]
sirius  \PYGZhy{}z 354.1347 \PYGZhy{}p orbitrap  \PYGZhy{}1 demo\PYGZhy{}data/txt/chelidonine\PYGZus{}ms.txt
        \PYGZhy{}2 demo\PYGZhy{}data/txt/chelidonine\PYGZus{}msms1.txt demo\PYGZhy{}data/txt/chelidonine\PYGZus{}msms2.txt

1.) C20H19NO5         score: 33.17    tree: +27.48    iso: 5.69       peaks: 13       95.44 \PYGZpc{}
2.) C16H22N2O5P       score: 32.35    tree: +26.77    iso: 5.58       peaks: 13       95.44 \PYGZpc{}
3.) C12H23N3O7S       score: 24.62    tree: +24.62    iso: 0.00       peaks: 13       95.44 \PYGZpc{}
4.) C18H17N4O4        score: 23.28    tree: +23.28    iso: 0.00       peaks: 14       95.79 \PYGZpc{}
5.) C14H20N5O4P       score: 21.61    tree: +21.61    iso: 0.00       peaks: 14       95.79 \PYGZpc{}
\end{Verbatim}

The overall score is the sum of the fragmentation pattern score and the isotope pattern score. If the isotope pattern score is negative, it is set to zero. If at least one isotope pattern score is greater than 10, the isotope pattern is considered to have \emph{good quality} and only the candidates with best isotope pattern scores are selected for further fragmentation pattern analysis.

If you want to analyze spectra measured with Orbitrap or FT-ICR, you should specify the appropriate analysis profile. A profile is a set of configuration options and scoring functions SIRIUS will use for its analysis. For example, the Orbitrap and FT-ICR profiles having tighter constraints for the allowed mass deviation but do not rely so much on the intensity of isotope peaks. You can set the profile with the \code{-p \textless{}name\textgreater{}} option. By default, qtof is used as profile.

SIRIUS recognizes the following options:
\index{command line option!-p \textless{}name\textgreater{}, --profile \textless{}name\textgreater{}}\index{-p \textless{}name\textgreater{}, --profile \textless{}name\textgreater{}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-p}\pysigline{\bfcode{-p}\code{~\textless{}name\textgreater{}}\code{,~}\bfcode{-{-}profile}\code{~\textless{}name\textgreater{}}}
Specify the used analysis profile. Choose either \code{qtof}, \code{orbitrap} or \code{fticr}. By default, \code{qtof} is selected.

\end{fulllineitems}

\index{command line option!-o \textless{}dirname\textgreater{}, --output \textless{}dirname\textgreater{}}\index{-o \textless{}dirname\textgreater{}, --output \textless{}dirname\textgreater{}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-o}\pysigline{\bfcode{-o}\code{~\textless{}dirname\textgreater{}}\code{,~}\bfcode{-{-}output}\code{~\textless{}dirname\textgreater{}}}
Specify the output directory. If given, SIRIUS will write the computed
fragmentation trees into this directory.

\end{fulllineitems}

\index{command line option!-O \textless{}format\textgreater{}, --format \textless{}format\textgreater{}}\index{-O \textless{}format\textgreater{}, --format \textless{}format\textgreater{}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-O}\pysigline{\bfcode{-O}\code{~\textless{}format\textgreater{}}\code{,~}\bfcode{-{-}format}\code{~\textless{}format\textgreater{}}}
Specify the format of the output of the fragmentation trees. This can be either \code{json} (machine readable), \code{dot} (for vizualization) or \code{sirius} (can be viewed with the SIRIUS Graphical User Interface).

\end{fulllineitems}

\index{command line option!-f {[}list of formulas{]}, --formula {[}list of formulas{]}}\index{-f {[}list of formulas{]}, --formula {[}list of formulas{]}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-f}\pysigline{\bfcode{-f}\code{~{[}list~of~formulas{]}}\code{,~}\bfcode{-{-}formula}\code{~{[}list~of~formulas{]}}}
Specify a list of candidate formulas (separated by whitespaces) that should be considered during analysis. This option is helpful if you performed a database search beforehand and only want to consider molecular formulas found in the database. It is recommended to first consider all molecular formulas (and omit this option) and filter the candidate list afterwards. However, specifying a subset of molecular formulas with this option might greatly improve the speed of the analysis especially for large molecules.

\end{fulllineitems}

\index{command line option!-a, --annotate}\index{-a, --annotate!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-a}\pysigline{\bfcode{-a}\code{}\code{,~}\bfcode{-{-}annotate}\code{}}
If set, SIRIUS will write the annotated spectrum containing the explanations (molecular formulas) for all identified peaks in a CSV file within the specified output directory.

\end{fulllineitems}

\index{command line option!-c \textless{}num\textgreater{}, --candidates \textless{}num\textgreater{}}\index{-c \textless{}num\textgreater{}, --candidates \textless{}num\textgreater{}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-c}\pysigline{\bfcode{-c}\code{~\textless{}num\textgreater{}}\code{,~}\bfcode{-{-}candidates}\code{~\textless{}num\textgreater{}}}
The number of candidates in the output. By default, SIRIUS will only write the five best candidates.

\end{fulllineitems}

\index{command line option!-s \textless{}val\textgreater{}, --isotope \textless{}val\textgreater{}}\index{-s \textless{}val\textgreater{}, --isotope \textless{}val\textgreater{}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-s}\pysigline{\bfcode{-s}\code{~\textless{}val\textgreater{}}\code{,~}\bfcode{-{-}isotope}\code{~\textless{}val\textgreater{}}}
This option specifies the way SIRIUS will handle the isotope patterns. If it is set to \code{omit}, SIRIUS will omit the isotope pattern analysis. If it is set to \code{filter}, SIRIUS will use the isotope pattern to select a subset of candidates before starting the fragmentation pattern analysis (this will improve the speed of the analysis). Only if it is set to \code{score}, SIRIUS will use it for filtering and scoring the candidates. The default setting is \code{score}.

\end{fulllineitems}

\index{command line option!-e \textless{}alphabet\textgreater{}, --elements \textless{}alphabet\textgreater{}}\index{-e \textless{}alphabet\textgreater{}, --elements \textless{}alphabet\textgreater{}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-e}\pysigline{\bfcode{-e}\code{~\textless{}alphabet\textgreater{}}\code{,~}\bfcode{-{-}elements}\code{~\textless{}alphabet\textgreater{}}}
Specify the used chemical alphabet. See {\hyperref[commandline:alphabets]{\crossref{\DUrole{std,std-ref}{Chemical Alphabets}}}}. By default, \code{CHNOP{[}5{]}S} is used.

\end{fulllineitems}

\index{command line option!-i \textless{}ion\textgreater{}, --ion \textless{}ion\textgreater{}}\index{-i \textless{}ion\textgreater{}, --ion \textless{}ion\textgreater{}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-i}\pysigline{\bfcode{-i}\code{~\textless{}ion\textgreater{}}\code{,~}\bfcode{-{-}ion}\code{~\textless{}ion\textgreater{}}}
Specify the used ionization. See {\hyperref[commandline:ions]{\crossref{\DUrole{std,std-ref}{Ion Modes}}}}. By default, \code{{[}M+H{]}+} is used.

\end{fulllineitems}

\index{command line option!-z \textless{}mz\textgreater{}, --parentmass \textless{}mz\textgreater{}}\index{-z \textless{}mz\textgreater{}, --parentmass \textless{}mz\textgreater{}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-z}\pysigline{\bfcode{-z}\code{~\textless{}mz\textgreater{}}\code{,~}\bfcode{-{-}parentmass}\code{~\textless{}mz\textgreater{}}}
Specify the parentmass of the input spectra. You have to give the exact measured value, not the selected ion mass.

\end{fulllineitems}

\index{command line option!-1 \textless{}file\textgreater{}, --ms1 \textless{}file\textgreater{}}\index{-1 \textless{}file\textgreater{}, --ms1 \textless{}file\textgreater{}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-1}\pysigline{\bfcode{-1}\code{~\textless{}file\textgreater{}}\code{,~}\bfcode{-{-}ms1}\code{~\textless{}file\textgreater{}}}
Specify the file path to the MS spectrum of the measured compound.

\end{fulllineitems}

\index{command line option!-2 \textless{}file\textgreater{}, --ms2 \textless{}file\textgreater{}}\index{-2 \textless{}file\textgreater{}, --ms2 \textless{}file\textgreater{}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-2}\pysigline{\bfcode{-2}\code{~\textless{}file\textgreater{}}\code{,~}\bfcode{-{-}ms2}\code{~\textless{}file\textgreater{}}}
Specify one or multiple file paths to the MS/MS spectra of the measured compound

\end{fulllineitems}

\index{command line option!--ppm-max \textless{}value\textgreater{}}\index{--ppm-max \textless{}value\textgreater{}!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption--ppm-max}\pysigline{\bfcode{-{-}ppm-max}\code{~\textless{}value\textgreater{}}}
Specify the allowed mass deviation of the fragment peaks in ppm. By default, Q-TOF instruments use 10 ppm and Orbitrap instruments use 5 ppm.

\end{fulllineitems}

\index{command line option!--auto-charge}\index{--auto-charge!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption--auto-charge}\pysigline{\bfcode{-{-}auto-charge}\code{}}
If this option is set, SIRIUS will annotate the fragment peaks with ion formulas instead of neutral molecular formulas. Use this option if you do not know the correct ionization.

\end{fulllineitems}

\index{command line option!--no-recalibrate}\index{--no-recalibrate!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption--no-recalibrate}\pysigline{\bfcode{-{-}no-recalibrate}\code{}}
If this option is set, SIRIUS will not recalibrate the spectrum during the analysis.

\end{fulllineitems}

\index{command line option!--fingerid}\index{--fingerid!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption--fingerid}\pysigline{\bfcode{-{-}fingerid}\code{}}
If this option is set, SIRIUS will search for molecular structure using CSI:FingerID after determining the molecular formula

\end{fulllineitems}

\index{command line option!--database, -d}\index{--database, -d!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption--database}\pysigline{\bfcode{-{-}database}\code{}\code{,~}\bfcode{-d}\code{}}
If this option is set, SIRIUS will only consider molecular formulas from the given database. Either \code{PubChem} or \code{bio} can be chosen, as well as other concrete databases like \code{hmdb}, \code{kegg}, \code{knapsack{}`{}`{}`or {}`{}`biocyc}. When used with the option --fingerid, the chosen database will also be used to search for candidate structures. When setting this value to \code{all}, SIRIUS will consider all possible molecular formulas (without database search) and CSI:FingerID will use PubChem.

\end{fulllineitems}

\index{command line option!--fingerid-db}\index{--fingerid-db!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption--fingerid-db}\pysigline{\bfcode{-{-}fingerid-db}\code{}}
With this option you can specify the structure database to search in with CSI:FingerID. You only have to set this option if you want to have different settings for searching molecular formulas and structures. For example, it might be common to search in the space of all possible molecular formulas (using -d all) but restrict the structures to a user specified database. 

\end{fulllineitems}

\index{command line option!-h, --help}\index{-h, --help!command line option}

\begin{fulllineitems}
\phantomsection\label{commandline:cmdoption-h}\pysigline{\bfcode{-h}\code{}\code{,~}\bfcode{-{-}help}\code{}}
display help

\end{fulllineitems}


See the following examples for running SIRIUS commandline tool:

\begin{Verbatim}[commandchars=\\\{\}]
sirius \PYGZhy{}p orbitrap \PYGZhy{}z 239.0315 \PYGZhy{}i [M+Na]+ \PYGZhy{}1 bergapten\PYGZus{}ms.csv
                \PYGZhy{}2 bergapten\PYGZus{}msms1.csv bergapten\PYGZus{}msms2.csv
sirius \PYGZhy{}p fticr \PYGZhy{}z 215.0350 \PYGZhy{}i [M\PYGZhy{}H]\PYGZhy{} \PYGZhy{}e CHNOPSCl[2] \PYGZhy{}c 10 \PYGZhy{}s omit
                \PYGZhy{}1 unknown\PYGZus{}ms1.csv \PYGZhy{}2 unknown\PYGZus{}ms2.csv
sirius \PYGZhy{}p qtof \PYGZhy{}z 215.035 \PYGZhy{}i 1\PYGZhy{} \PYGZhy{}\PYGZhy{}auto\PYGZhy{}charge \PYGZhy{}2 unknown\PYGZus{}ms2.csv
sirius \PYGZhy{}c 10 \PYGZhy{}o trees \PYGZhy{}O json msdir
sirius \PYGZhy{}f C6H12O6 C5H6N7O C7H16OS2 \PYGZhy{}i [M+H]+ \PYGZhy{}1 ms.csv \PYGZhy{}2 msms.csv
\end{Verbatim}


\paragraph{Computing fragmentation trees}
\label{commandline:computing-fragmentation-trees}
If you already know the correct molecular formula and just want to compute a
fragmentation tree, you can specify a single molecular formula with
the \code{-f} option. SIRIUS will then only compute a tree for this
molecular formula. If your input data is in \code{.ms} format, the molecular
formula might be already specified within the file. If a molecular formula is
specified, the parent mass can be omitted. However, you still have to specify
the ionization (except for default value \code{{[}M+H{]}+}):

\begin{Verbatim}[commandchars=\\\{\}]
sirius \PYGZhy{}f C20H19NO5 \PYGZhy{}2 demo\PYGZhy{}data/txt/chelidonine\PYGZus{}msms2.txt demo\PYGZhy{}data/txt/chelidonine\PYGZus{}msms2.txt
\end{Verbatim}


\paragraph{Visualizing fragmentation trees}
\label{commandline:visualizing-fragmentation-trees}
SIRIUS supports three output formats for fragmentation trees: dot (graphviz format), json (machine readable format), and sirius (can be viewed with the SIRIUS Graphical User Interface). The commandline tool Graphviz \footnote{\sphinxAtStartFootnote%
\url{http://www.graphviz.org/}
} can transform dot files into image formats (PDF, SVG, PNG etc.). After installing Graphviz you can display tree files as follows:

\begin{Verbatim}[commandchars=\\\{\}]
sirius \PYGZhy{}p orbitrap \PYGZhy{}f C20H17NO6 \PYGZhy{}o trees demo\PYGZhy{}data/ms/Bicuculline.ms
dot \PYGZhy{}Tpdf \PYGZhy{}O trees/Bicuculline.dot
\end{Verbatim}

This creates a file Bicuculline.dot.pdf:

\begin{center}
  \includegraphics[width=0.3\textwidth]{{tree}.pdf}
\end{center}

SIRIUS uses automatically the file name of the input spectrum to name the
output file. You can specify another filename with the \code{-o} option, as
long as only one fragmentation tree is computed.
\begin{quote}
sirius -p orbitrap -f C20H17NO6 -o compound.dot demo-data/ms/Bicuculline.ms
dot -Tpdf -O compound.dot
\end{quote}

\paragraph{Identifying Molecular Structures}
\label{commandline:identifying-molecular-structures}

You can also search for molecular structures with
CSI:FingerID. Just add the option \code{--fingerid} to trigger a CSI:FingerID
search after fragmentation tree computation. With \code{--database} can now
also specify the database SIRIUS should search in. Available
are \code{pubchem} and \code{bio}.  You can also specify certain databases
like \emph{code} and \emph{code}; however, we strongly recommended to instead
filter the output list afterwards.

SIRIUS will generate CSV files for each input spectrum containing a ordered candidate list of structures with the CSI:FingerID score. Furthermore, another result CSV file is generated only the top candidates from all input spectra ordered by their confidence.
\begin{quote}

sirius -c 10 --database=pubchem --fingerid demo-data/ms/Bicuculline.ms
\end{quote}


\paragraph{Demo Data}
\label{commandline:demo-data}

You can download some sample spectra from the SIRIUS website
at \url{http://bio.informatik.uni-jena.de/sirius2/wp-content/uploads/2015/05/demo.zip}

The demo-data contain examples for three different data formats readable by
SIRIUS. The MGF folder contain an example for a MGF file containing a single
compound with several MS/MS spectra measured on an Orbitrap
instrument. SIRIUS recognizes that these MS/MS spectra belong to the same
compound because they have the same parent mass. To analyze this compound,
run:
%
\begin{Verbatim}[commandchars=\\\{\}]
sirius \PYGZhy{}p orbitrap demo\PYGZhy{}data/mgf/laudanosine.mgf
\end{Verbatim}

The output is:
%
\begin{Verbatim}[commandchars=\\\{\}]
1.) C21H27NO4         score: 25.41    tree: +17.55    iso: 7.86       peaks: 12       97.94 \PYGZpc{}
2.) C17H30N2O4P       score: 21.46    tree: +13.97    iso: 7.49       peaks: 12       97.94 \PYGZpc{}
3.) C15H28N5O3P       score: 15.00    tree: +15.00    iso: 0.00       peaks: 11       87.04 \PYGZpc{}
4.) C19H25N4O3        score: 14.66    tree: +14.66    iso: 0.00       peaks: 11       87.16 \PYGZpc{}
5.) C14H27N7O2S       score: 13.69    tree: +13.69    iso: 0.00       peaks: 11       97.38 \PYGZpc{}
\end{Verbatim}

This is a ranking list of the top molecular formula candidates. The best
candidate is \ce{C21H27NO4} with a overall score of 25.41. This score is the
sum of the fragmentation pattern scoring (17.55) and the isotope pattern
scoring (7.86). For the last three candidates, the isotope pattern scoring is
0. In fact, this score can never fall below zero. If all isotope pattern
scores are zero, you can assume that the isotope pattern has very low quality
and cannot be used to determine the molecular formula. If the isotope pattern
score of the top candidate is over 10, it is assumed to be a high quality
isotope pattern. In this case, the isotope pattern is also used to filter out
unlikely candidates and speed up the analysis.

The last two columns contain the number of explained peaks in MS/MS spectrum
as well as the relative amount of explained intensity. The last value should
usually be over 80 \% or even 90 \%. If this value is very low you either
have strange high intensive noise in your spectrum, or the allowed mass
deviation might be too low to explain all the peaks.

If you want to look at the fragmentation trees, you have to add the output
option:

\begin{Verbatim}[commandchars=\\\{\}]
sirius \PYGZhy{}p orbitrap \PYGZhy{}o outputdir demo\PYGZhy{}data/mgf/laudanosine.mgf
\end{Verbatim}

Now, SIRIUS will write the computed fragmentation trees into
the \code{outputdir} directory. You can visualize this trees in PDF format
using Graphviz:

\begin{Verbatim}[commandchars=\\\{\}]
dot \PYGZhy{}Tpdf \PYGZhy{}O outputdir/laudanosine\PYGZus{}1\PYGZus{}C21H27NO4.dot
\end{Verbatim}

This creates a pdf file \code{outputdir/laudanosine\_1\_C21H27NO4.dot.pdf}.

The directory \code{ms} contains two examples of the ms format. Each file contains a single compound measured with an Orbitrap instrument. To analyze this compound run:

\begin{Verbatim}[commandchars=\\\{\}]
sirius \PYGZhy{}p orbitrap \PYGZhy{}o outputdir demo\PYGZhy{}data/ms/Bicuculline.ms
\end{Verbatim}

As the ms file already contains the correct molecular formula, SIRIUS will
directly compute the fragmentation tree. For such cases (as well as when you
specify exactly one molecular formula via \code{-f} option) you can also
specify the concrete filename of the output file:

\begin{Verbatim}[commandchars=\\\{\}]
sirius \PYGZhy{}p orbitrap \PYGZhy{}o mycompound.dot demo\PYGZhy{}data/ms/Bicuculline.ms
\end{Verbatim}

If you want to enforce a molecular formula analysis and ranking (although the correct molecular formula is given within the file) you can specify the number of candidates with the \code{-c} option:

\begin{Verbatim}[commandchars=\\\{\}]
sirius \PYGZhy{}p orbitrap \PYGZhy{}c 5 demo\PYGZhy{}data/ms/Bicuculline.ms
\end{Verbatim}

SIRIUS will now ignore the correct molecular formula in the file and output the 5 best candidates.

The TXT folder contains simple peaklist files. Such file formats can be easily extracted from Excel spreadsheets. However, they do not contain meta information like the MS level and the parent mass. So you have to specify this information via commandline options:

\begin{Verbatim}[commandchars=\\\{\}]
sirius  \PYGZhy{}p orbitrap  \PYGZhy{}z 354.134704589844 \PYGZhy{}1 demo\PYGZhy{}data/txt/chelidonine\PYGZus{}ms.txt
        \PYGZhy{}2 demo\PYGZhy{}data/txt/chelidonine\PYGZus{}msms1.txt demo\PYGZhy{}data/txt/chelidonine\PYGZus{}msms2.txt
\end{Verbatim}

The demo data contain a clean MS spectrum (e.g. there is only one isotope pattern contained in the MS spectrum). In such cases, SIRIUS can infer the correct parent mass from the MS data (by simply using the monoisotopic mass of the isotope pattern as parent mass). So you can omit the \code{-z} option in this cases.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%  CHAPTER  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\chapter{SIRIUS Java Library}
\label{library:sirius-java-library}\label{library::doc}

You can integrate the SIRIUS library in your Java project, either by using
Maven \footnote{\sphinxAtStartFootnote\url{https://maven.apache.org/}} or by
including the jar file directly. The latter is not recommended, as the
SIRIUS jar contains also dependencies to other external libraries.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Maven Integration}
\label{library:maven-integration}
Add the following repository to your pom file:

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{n+nt}{\PYGZlt{}distributionManagement}\PYG{n+nt}{\PYGZgt{}}
  \PYG{n+nt}{\PYGZlt{}repository}\PYG{n+nt}{\PYGZgt{}}
      \PYG{n+nt}{\PYGZlt{}id}\PYG{n+nt}{\PYGZgt{}}bioinf\PYGZhy{}jena\PYG{n+nt}{\PYGZlt{}/id\PYGZgt{}}
      \PYG{n+nt}{\PYGZlt{}name}\PYG{n+nt}{\PYGZgt{}}bioinf\PYGZhy{}jena\PYGZhy{}releases\PYG{n+nt}{\PYGZlt{}/name\PYGZgt{}}
      \PYG{n+nt}{\PYGZlt{}url}\PYG{n+nt}{\PYGZgt{}}http://bio.informatik.uni\PYGZhy{}jena.de/artifactory/libs\PYGZhy{}releases\PYGZhy{}local\PYG{n+nt}{\PYGZlt{}/url\PYGZgt{}}
  \PYG{n+nt}{\PYGZlt{}/repository\PYGZgt{}}
\PYG{n+nt}{\PYGZlt{}/distributionManagement\PYGZgt{}}
\end{Verbatim}

Now you can integrate SIRIUS in your project by adding the following dependency:

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{n+nt}{\PYGZlt{}dependency}\PYG{n+nt}{\PYGZgt{}}
  \PYG{n+nt}{\PYGZlt{}groupId}\PYG{n+nt}{\PYGZgt{}}de.unijena.bioinf\PYG{n+nt}{\PYGZlt{}/groupId\PYGZgt{}}
  \PYG{n+nt}{\PYGZlt{}artifactId}\PYG{n+nt}{\PYGZgt{}}sirius_api\PYG{n+nt}{\PYGZlt{}/artifactId\PYGZgt{}}
  \PYG{n+nt}{\PYGZlt{}version}\PYG{n+nt}{\PYGZgt{}}3.5.0\PYG{n+nt}{\PYGZlt{}/version\PYGZgt{}}
\PYG{n+nt}{\PYGZlt{}/dependency\PYGZgt{}}
\end{Verbatim}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Main API}
\label{library:main-api}

The main class in SIRIUS is \textbf{de.unijena.bioinf.sirius.SIRIUS}. It is
basically a wrapper around the important functionalities of the
library. Although there are special classes for all parts of the analysis
pipeline it is recommended to only use the SIRIUS class as the API of all
other classes might change in future releases. The SIRIUS class also
provides factory methods for the most important data structures. Although,
for many of this data structures you could also use their constructors
directly, it is recommended to use the methods in the SIRIUS class.

\phantomsection\label{library:package-de.unijena.bioinf.sirius}\index{de.unijena.bioinf.sirius (package)}\index{SIRIUS (Java class)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.SIRIUS}\pysigline{public class \bfcode{SIRIUS}}
The main class in SIRIUS. Provides the basic functionality of the method.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{profile}} -- the profile name. Can be one of `qtof', `orbitrap' or `fticr'. If omitted, the default profile (`qtof') is used.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

The main functions of SIRIUS are either identifying the molecular formula
of a compound, or computing a fragmentation tree for a given molecular formula
and MS/MS spectrum. The SIRIUS class provides two methods for this purpose:
identify and compute. The basic input type is an Ms2Experiment. It can be
seen as a set of MS/MS spectra derived from the same precursor as well as a
MS spectrum containing this precursor peak. The output of SIRIUS is an
instance of IdentificationResult, containing the score and the corresponding
fragmentation tree for the candidate molecular formula.


\subsection{Create datastructures}
\label{library:create-datastructures}
SIRIUS provides the following functions to create the basic data structures:
\index{wrapSpectrum(double{[}{]}, double{[}{]}) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.wrapSpectrum(double__, double__)}\pysiglinewithargsret{public Spectrum\textless{}Peak\textgreater{} \bfcode{wrapSpectrum}}{double{[}{]}\emph{ mz}, double{[}{]}\emph{ intensities}}{}
Wraps an array of m/z values and and array of intensity values into a spectrum object that can be used by the SIRIUS library. The resulting spectrum is a lightweight view on the array, so changes in the array are reflected in the spectrum. The spectrum object itself is immutable.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{mz}} -- mass to charge ratios

\item {} 
\textbf{\texttt{intensities}} -- intensity values. Can be normalized or absolute values - SIRIUS will normalize them itself if necessary

\end{itemize}

\item[{Returns}] \leavevmode
view on the arrays implementing the Spectrum interface

\end{description}\end{quote}

\end{fulllineitems}

\index{getElement(String) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.getElement(String)}\pysiglinewithargsret{public Element \bfcode{getElement}}{\href{http://docs.oracle.com/javase/6/docs/api/java/lang/String.html}{String}\emph{ symbol}}{}
Lookup the symbol in the periodic table and returns the corresponding Element object or null if no element with this symbol exists.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{symbol}} -- symbol of the element, e.g. H for hydrogen or Cl for chlorine

\end{itemize}

\item[{Returns}] \leavevmode
instance of Element class

\end{description}\end{quote}

\end{fulllineitems}

\index{getIonization(String) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.getIonization(String)}\pysiglinewithargsret{public Ionization \bfcode{getIonization}}{\href{http://docs.oracle.com/javase/6/docs/api/java/lang/String.html}{String}\emph{ name}}{}
Lookup the ionization name and returns the corresponding ionization object or null if no ionization with this name is registered. The name of an ionization has the syntax {[}M+ADDUCT{]}CHARGE, for example {[}M+H{]}+ or {[}M-H{]}-.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{name}} -- name of the ionization

\end{itemize}

\item[{Returns}] \leavevmode
Adduct instance

\end{description}\end{quote}

\end{fulllineitems}

\index{getCharge(int) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.getCharge(int)}\pysiglinewithargsret{public Charge \bfcode{getCharge}}{int\emph{ charge}}{}
Charges are subclasses of Ionization. So they can be used everywhere as replacement for ionizations. A charge is very similar to the {[}M{]}+ and {[}M{]}- ionizations. However, the difference is that {[}M{]}+ describes an intrinsically charged compound where the Charge +1 describes a compound with unknown adduct.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{charge}} -- either 1 for positive or -1 for negative charges.

\end{itemize}

\item[{Returns}] \leavevmode
a Charge instance which is also a subclass of Ionization

\end{description}\end{quote}

\end{fulllineitems}

\index{getMassDeviation(int, double) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.getMassDeviation(int, double)}\pysiglinewithargsret{public Deviation \bfcode{getMassDeviation}}{int\emph{ ppm}, double\emph{ abs}}{}
Creates a Deviation object that describes a mass deviation as maximum of a relative term (in ppm) and an absolute term. Usually, mass accuracy is given as relative term in ppm, as measurement errors increase with higher masses. However, for very small compounds (and fragments!) these relative values might overestimate the mass accuracy. Therefore, an absolute value have to be given.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{ppm}} -- mass deviation as relative value (in ppm)

\item {} 
\textbf{\texttt{abs}} -- mass deviation as absolute value (m/z)

\end{itemize}

\item[{Returns}] \leavevmode
Deviation object

\end{description}\end{quote}

\end{fulllineitems}

\index{parseFormula(String) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.parseFormula(String)}\pysiglinewithargsret{ MolecularFormula \bfcode{parseFormula}}{\href{http://docs.oracle.com/javase/6/docs/api/java/lang/String.html}{String}\emph{ f}}{}
Parses a molecular formula from the given string
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{f}} -- molecular formula (e.g. in Hill notation)

\end{itemize}

\item[{Returns}] \leavevmode
immutable molecular formula object

\end{description}\end{quote}

\end{fulllineitems}

\index{getMs2Experiment(MolecularFormula, Ionization, Spectrum, Spectrum) (Java method)}\index{getMs2Experiment(double, Ionization, Spectrum, Spectrum) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.getMs2Experiment(MolecularFormula, Ionization, Spectrum, Spectrum)}\pysiglinewithargsret{public Ms2Experiment \bfcode{getMs2Experiment}}{MolecularFormula\emph{ formula}, Ionization\emph{ ion}, Spectrum\textless{}Peak\textgreater{}\emph{ ms1}, Spectrum...\emph{ ms2}}{}\phantomsection\label{library:de.unijena.bioinf.sirius.getMs2Experiment(double, Ionization, Spectrum, Spectrum)}\pysiglinewithargsret{public Ms2Experiment \bfcode{getMs2Experiment}}{double\emph{ parentmass}, Ionization\emph{ ion}, Spectrum\textless{}Peak\textgreater{}\emph{ ms1}, Spectrum...\emph{ ms2}}{}
Creates a Ms2Experiment object from the given MS and MS/MS spectra. A Ms2Experiment is NOT a single run or measurement, but a measurement of a concrete compound. So a MS spectrum might contain several Ms2Experiments. However, each MS/MS spectrum should have on precursor or parent mass. All MS/MS spectra with the same precursor together with the MS spectrum containing this precursor peak can be seen as one Ms2Experiment.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{formula}} -- neutral molecular formula of the compound

\item {} 
\textbf{\texttt{parentmass}} -- if neutral molecular formula is unknown, you have to provide the ion mass

\item {} 
\textbf{\texttt{ion}} -- ionization mode (can be an instance of Charge if the exact adduct is unknown)

\item {} 
\textbf{\texttt{ms1}} -- the MS spectrum containing the isotope pattern of the measured compound. Might be null

\item {} 
\textbf{\texttt{ms2}} -- a list of MS/MS spectra containing the fragmentation pattern of the measured compound

\end{itemize}

\item[{Returns}] \leavevmode
a MS2Experiment instance, ready to be analyzed by SIRIUS

\end{description}\end{quote}

\end{fulllineitems}

\index{getFormulaConstraints(String) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.getFormulaConstraints(String)}\pysiglinewithargsret{public FormulaConstraints \bfcode{getFormulaConstraints}}{\href{http://docs.oracle.com/javase/6/docs/api/java/lang/String.html}{String}\emph{ constraints}}{}
Formula Constraints consist of a chemical alphabet (a subset of the periodic table, determining which elements might occur in the measured compounds) and upper bounds for each of this elements. A formula constraint can be given like a molecular formula. Upper bounds are written in square brackets or omitted, if any number of this element should be allowed.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{constraints}} -- string representation of the constraint, e.g. ``CHNOP{[}5{]}S{[}20{]}''

\end{itemize}

\item[{Returns}] \leavevmode
formula constraint object

\end{description}\end{quote}

\end{fulllineitems}

\index{parseExperiment(File) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.parseExperiment(File)}\pysiglinewithargsret{public CloseableIterator\textless{}Ms2Experiment\textgreater{} \bfcode{parseExperiment}}{File\emph{ file}}{}
parses a file and return an iterator over all MS/MS compounds contained in this file.
a compound consists of all MS and MS/MS spectra belonging to one feature (=compound). Supported file formats are .ms and .mgf. The returned iterator supports the close method to close the input stream. The stream is closed automatically, after iterating the last element. However, it is recommended to use the try-close syntax from Java 7 to ensure a proper closing of the stream.

\end{fulllineitems}



\subsection{Provided algorithms}
\label{library:provided-algorithms}\index{identify(Ms2Experiment, int, boolean, IsotopePatternHandling, Set) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.identify(Ms2Experiment, int, boolean, IsotopePatternHandling, Set)}\pysiglinewithargsret{ List\textless{}{\hyperref[library:de.unijena.bioinf.sirius.IdentificationResult]{\crossref{IdentificationResult}}}\textgreater{} \bfcode{identify}}{Ms2Experiment\emph{ uexperiment}, int\emph{ numberOfCandidates}, boolean\emph{ recalibrating}, IsotopePatternHandling\emph{ deisotope}, Set\textless{}MolecularFormula\textgreater{}\emph{ whiteList}}{}
Identify the molecular formula of the measured compound by combining an isotope pattern analysis on MS data with a fragmentation pattern analysis on MS/MS data
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{uexperiment}} -- input data

\item {} 
\textbf{\texttt{numberOfCandidates}} -- number of candidates to output

\item {} 
\textbf{\texttt{recalibrating}} -- true if spectra should be recalibrated
during fragmentation tree computation

\item {} 
\textbf{\texttt{deisotope}} -- set this to `omit' to ignore isotope pattern, `filter' to use it for selecting molecular formula candidates or `score' to rerank the candidates according to their isotope pattern

\item {} 
\textbf{\texttt{whiteList}} -- restrict the analysis to this subset of molecular formulas. If this set is empty, consider all possible molecular formulas

\end{itemize}

\item[{Returns}] \leavevmode
a list of identified molecular formulas together with their fragmentation trees

\end{description}\end{quote}

\end{fulllineitems}

\index{compute(Ms2Experiment, MolecularFormula, boolean) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.compute(Ms2Experiment, MolecularFormula, boolean)}\pysiglinewithargsret{public {\hyperref[library:de.unijena.bioinf.sirius.IdentificationResult]{\crossref{IdentificationResult}}} \bfcode{compute}}{Ms2Experiment\emph{ experiment}, MolecularFormula\emph{ formula}, boolean\emph{ recalibrating}}{}
Compute a fragmentation tree for the given MS/MS data using the given neutral molecular formula as explanation for the measured compound
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{experiment}} -- input data

\item {} 
\textbf{\texttt{formula}} -- neutral molecular formula of the measured compound

\item {} 
\textbf{\texttt{recalibrating}} -- true if spectra should be recalibrated
during fragmentation tree computation

\end{itemize}

\item[{Returns}] \leavevmode
A single instance of IdentificationResult containing the computed fragmentation tree

\end{description}\end{quote}

\end{fulllineitems}

\index{decompose(double, Ionization, FormulaConstraints, Deviation) (Java method)}\index{decompose(double, Ionization, FormulaConstraints) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.decompose(double, Ionization, FormulaConstraints, Deviation)}\pysiglinewithargsret{public List\textless{}MolecularFormula\textgreater{} \bfcode{decompose}}{double\emph{ mass}, Ionization\emph{ ion}, FormulaConstraints\emph{ constr}, Deviation\emph{ dev}}{}\phantomsection\label{library:de.unijena.bioinf.sirius.decompose(double, Ionization, FormulaConstraints)}\pysiglinewithargsret{public List\textless{}MolecularFormula\textgreater{} \bfcode{decompose}}{double\emph{ mass}, Ionization\emph{ ion}, FormulaConstraints\emph{ constr}}{}
Decomposes a mass and return a list of all molecular formulas which ionized mass is near the measured mass.
The maximal distance between the neutral mass of the measured ion and the theoretical mass of the decomposed formula depends on the chosen profile. For qtof it is 10 ppm, for Orbitrap and FTICR it is 5 ppm.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{mass}} -- mass of the measured ion

\item {} 
\textbf{\texttt{ion}} -- ionization mode (might be a Charge, in which case the decomposer will enumerate the ion formulas instead of the neutral formulas)

\item {} 
\textbf{\texttt{constr}} -- the formula constraints, defining the allowed elements and their upperbounds

\item {} 
\textbf{\texttt{dev}} -- the allowed mass deviation of the measured ion from the theoretical ion masses

\end{itemize}

\item[{Returns}] \leavevmode
list of molecular formulas which theoretical ion mass is near the given mass

\end{description}\end{quote}

\end{fulllineitems}

\index{simulateIsotopePattern(MolecularFormula, Ionization) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.simulateIsotopePattern(MolecularFormula, Ionization)}\pysiglinewithargsret{public Spectrum\textless{}Peak\textgreater{} \bfcode{simulateIsotopePattern}}{MolecularFormula\emph{ compound}, Ionization\emph{ ion}}{}
Simulates an isotope pattern for the given molecular formula and the chosen ionization
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{compound}} -- neutral molecular formula

\item {} 
\textbf{\texttt{ion}} -- ionization mode (might be a Charge)

\end{itemize}

\item[{Returns}] \leavevmode
spectrum containing the theoretical isotope pattern of this compound

\end{description}\end{quote}

\end{fulllineitems}



\subsection{Output type}
\label{library:package-de.unijena.bioinf.sirius}\label{library:output-type}\index{de.unijena.bioinf.sirius (package)}\index{IdentificationResult (Java class)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.IdentificationResult}\pysigline{public class \bfcode{IdentificationResult}}
The compute and identify methods return instances of
IdentificationResult. This class wraps a fragmentation tree and its scores. You can write the tree to a file using the writeTreeToFile method.

\end{fulllineitems}

\index{writeTreeToFile(File) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.writeTreeToFile(File)}\pysiglinewithargsret{public void \bfcode{writeTreeToFile}}{File\emph{ target}}{}
Writes the fragmentation tree into a file. The file format is determined by the file ending (either `.dot' or `.json')
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{target}} -- file name

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{writeAnnotatedSpectrumToFile(File) (Java method)}

\begin{fulllineitems}
\phantomsection\label{library:de.unijena.bioinf.sirius.writeAnnotatedSpectrumToFile(File)}\pysiglinewithargsret{public void \bfcode{writeAnnotatedSpectrumToFile}}{File\emph{ target}}{}
Writes the annotated spectrum into a csv file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{target}} -- file name

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%  CHAPTER  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newpage
\chapter{Frequently asked questions (FAQ)}
\input{../build/faq.tex}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%  CHAPTER  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newpage
\chapter{Changelog}
\input{../build/changelog.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%  BIBLIOGRAPHY  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\chapter{Bibliography}
\printbibliography[heading=none]


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\renewcommand{\indexname}{Index}
\printindex
\end{document}
